--- a/net/minecraft/entity/decoration/AbstractDecorationEntity.java
+++ b/net/minecraft/entity/decoration/AbstractDecorationEntity.java
@@ -31,7 +31,7 @@
     };
     private int obstructionCheckCounter;
     protected BlockPos attachmentPos;
-    protected Direction facing;
+    public Direction facing; // Loom :: public accessor
 
     protected AbstractDecorationEntity(EntityType<? extends AbstractDecorationEntity> entitytype, World world) {
         super(entitytype, world);
@@ -45,7 +45,7 @@
 
     protected void initDataTracker() {}
 
-    protected void setFacing(Direction direction) {
+    public void setFacing(Direction direction) { // Loom :: public accessor
         Validate.notNull(direction);
         Validate.isTrue(direction.getAxis().isHorizontal());
         this.facing = direction;
