--- a/net/minecraft/world/entity/animal/AbstractFish.java
+++ b/net/minecraft/world/entity/animal/AbstractFish.java
@@ -101,7 +101,7 @@
         Predicate predicate = EntitySelector.NO_SPECTATORS;
 
         predicate.getClass();
-        goalselector.addGoal(2, new AvoidEntityGoal<>(this, Player.class, 8.0F, 1.6D, 1.4D, predicate::test));
+        goalselector.addGoal(2, new AvoidEntityGoal<Player>(this, Player.class, 8.0F, 1.6D, 1.4D, predicate::test)); // Loom :: decompile error
         this.goalSelector.addGoal(4, new AbstractFish.FishSwimGoal(this));
     }
 
@@ -150,7 +150,7 @@
 
     }
 
-    protected abstract ItemStack getBucketItemStack();
+    public abstract ItemStack getBucketItemStack(); // Loom :: protected -> public (AbstractFishImpl)
 
     protected boolean canRandomSwim() {
         return true;
