--- a/net/minecraft//world/entity/animal/Animal.java
+++ b/net/minecraft//world/entity/animal/Animal.java
@@ -32,7 +32,7 @@
 
     static final int PARENT_AGE_AFTER_BREEDING = 6000;
     private int inLove;
-    private UUID loveCause;
+    public UUID loveCause; // Loom :: private -> public
 
     protected Animal(EntityType<? extends Animal> entitytype, Level level) {
         super(entitytype, level);
