--- a/net/minecraft//world/entity/animal/Parrot.java
+++ b/net/minecraft//world/entity/animal/Parrot.java
@@ -71,9 +71,11 @@
             return mob != null && Parrot.MOB_SOUND_MAP.containsKey(mob.getType());
         }
 
+        /* // Loom start :: decompile error
         public boolean test(@Nullable Object object) {
             return this.test((Mob) object);
         }
+         */ // Loom end
     };
     private static final Item POISONOUS_FOOD = Items.COOKIE;
     private static final Set<Item> TAME_FOOD = Sets.newHashSet(new Item[]{Items.WHEAT_SEEDS, Items.MELON_SEEDS, Items.PUMPKIN_SEEDS, Items.BEETROOT_SEEDS});
@@ -134,7 +136,7 @@
     public SpawnGroupData finalizeSpawn(ServerLevelAccessor serverlevelaccessor, DifficultyInstance difficultyinstance, MobSpawnType mobspawntype, @Nullable SpawnGroupData spawngroupdata, @Nullable CompoundTag compoundtag) {
         this.setVariant(this.random.nextInt(5));
         if (spawngroupdata == null) {
-            spawngroupdata = new AgeableMob.AgeableMobGroupData(false);
+            spawngroupdata = new AgeableMobGroupData(false);
         }
 
         return super.finalizeSpawn(serverlevelaccessor, difficultyinstance, mobspawntype, (SpawnGroupData) spawngroupdata, compoundtag);
@@ -339,7 +341,7 @@
         this.playSound(SoundEvents.PARROT_STEP, 0.15F, 1.0F);
     }
 
-    protected boolean isFlapping() {
+    public boolean isFlapping() { // Loom :: protected -> public (EntityImpl)
         return this.flyDist > this.nextFlap;
     }
 
