--- a/net/minecraft//world/entity/animal/Pufferfish.java
+++ b/net/minecraft//world/entity/animal/Pufferfish.java
@@ -31,7 +31,7 @@
 
     private static final EntityDataAccessor<Integer> PUFF_STATE = SynchedEntityData.defineId(Pufferfish.class, EntityDataSerializers.INT);
     int inflateCounter;
-    int deflateTimer;
+    public int deflateTimer; // Loom :: package-private -> public (PufferfishImpl)
     private static final Predicate<LivingEntity> SCARY_MOB = (livingentity) -> {
         return livingentity instanceof Player && ((Player) livingentity).isCreative() ? false : livingentity.getType() == EntityType.AXOLOTL || livingentity.getMobType() != MobType.WATER;
     };
