--- a/net/minecraft//world/entity/animal/Pufferfish.java
+++ b/net/minecraft//world/entity/animal/Pufferfish.java
@@ -30,7 +30,7 @@
 
     private static final EntityDataAccessor<Integer> PUFF_STATE = SynchedEntityData.defineId(Pufferfish.class, EntityDataSerializers.INT);
     private int inflateCounter;
-    private int deflateTimer;
+    public int deflateTimer; // Loom :: private -> public (PufferfishImpl)
     private static final Predicate<LivingEntity> NO_SPECTATORS_AND_NO_WATER_MOB = (livingentity) -> {
         return livingentity == null ? false : (livingentity instanceof Player && (livingentity.isSpectator() || ((Player) livingentity).isCreative()) ? false : livingentity.getMobType() != MobType.WATER);
     };
