--- a/net/minecraft/item/ArmorStandItem.java
+++ b/net/minecraft/item/ArmorStandItem.java
@@ -35,6 +35,11 @@
             ArmorStandEntity armorstandentity = (ArmorStandEntity) EntityType.ARMOR_STAND.create(world, itemstack.getTag(), (Text) null, itemusagecontext.getPlayer(), blockpos, SpawnReason.SPAWN_EGG, true, true);
 
             if (world.doesNotCollide((Entity) armorstandentity) && world.getEntities(armorstandentity, armorstandentity.getBoundingBox()).isEmpty()) {
+                // Loom start :: fire ArmorStandPlacedEvent
+                if (org.loomdev.loom.event.LoomEventDispatcher.onArmorStandPlaced(armorstandentity, blockpos, itemplacementcontext.getPlayer()).isCancelled()) {
+                    return ActionResult.FAIL;
+                }
+                // Loom end
                 if (!world.isClient) {
                     float f = (float) MathHelper.floor((MathHelper.wrapDegrees(itemusagecontext.getPlayerYaw() - 180.0F) + 22.5F) / 45.0F) * 45.0F;
 
