--- a/net/minecraft/item/BucketItem.java
+++ b/net/minecraft/item/BucketItem.java
@@ -118,14 +118,24 @@
                 int j = blockpos.getY();
                 int k = blockpos.getZ();
 
+                // Loom start :: fire BlockEvaporatedEvent
+                org.loomdev.loom.event.LoomEventDispatcher.onBlockEvaporated(world, blockpos, playerentity).thenAccept(event -> {
+                    if (event.isCancelled()) {
+                        // Place the water anyway if this even is cancelled
+                        world.setBlockState(blockpos, this.fluid.getDefaultState().getBlockState(), 11);
+                        this.playEmptyingSound(playerentity, world, blockpos);
+                        return;
+                    }
+
                 world.playSound(playerentity, blockpos, SoundEvents.BLOCK_FIRE_EXTINGUISH, SoundCategory.BLOCKS, 0.5F, 2.6F + (world.random.nextFloat() - world.random.nextFloat()) * 0.8F);
 
                 for (int l = 0; l < 8; ++l) {
                     world.addParticle(ParticleTypes.LARGE_SMOKE, (double) i + Math.random(), (double) j + Math.random(), (double) k + Math.random(), 0.0D, 0.0D, 0.0D);
                 }
-
+                });
+                // Loom end
                 return true;
-            } else if (block instanceof FluidFillable && this.fluid == Fluids.WATER) {
+            } else if (block instanceof FluidFillable && this.fluid == Fluids.WATER) { // TODO waterlog event
                 ((FluidFillable) block).tryFillWithFluid(world, blockpos, blockstate, ((FlowableFluid) this.fluid).getStill(false));
                 this.playEmptyingSound(playerentity, world, blockpos);
                 return true;
