--- a/net/minecraft/item/ItemStack.java
+++ b/net/minecraft/item/ItemStack.java
@@ -157,6 +157,17 @@
         BlockPos blockpos = itemusagecontext.getBlockPos();
         CachedBlockPosition cachedblockposition = new CachedBlockPosition(itemusagecontext.getWorld(), blockpos, false);
 
+        // Loom start :: fire BlockPlacedEvent and PlantFertilizedEvent
+        if (playerentity != null) {
+            if (item != Items.BONE_MEAL && org.loomdev.loom.event.LoomEventDispatcher.onBlockPlaced(itemusagecontext.getWorld(), blockpos, playerentity).isCancelled()) {
+                ((org.loomdev.loom.entity.player.PlayerImpl) playerentity.getLoomEntity()).updateInventory();
+                return ActionResult.FAIL;
+            } else if (item == Items.BONE_MEAL && org.loomdev.loom.event.LoomEventDispatcher.onPlantFertilized(itemusagecontext.getWorld(), blockpos, playerentity).isCancelled()) {
+                return ActionResult.FAIL;
+            }
+        }
+        // Loom end
+
         if (playerentity != null && !playerentity.abilities.allowModifyWorld && !this.canPlaceOn(itemusagecontext.getWorld().getTagManager(), cachedblockposition)) {
             return ActionResult.PASS;
         } else {
