--- a/net/minecraft/network/packet/s2c/play/PlayerListHeaderS2CPacket.java
+++ b/net/minecraft/network/packet/s2c/play/PlayerListHeaderS2CPacket.java
@@ -9,8 +9,8 @@
 
 public class PlayerListHeaderS2CPacket implements Packet<ClientPlayPacketListener> {
 
-    private Text header;
-    private Text footer;
+    public Text header; // Loom :: public access
+    public Text footer; // Loom :: public access
 
     public PlayerListHeaderS2CPacket() {}
 
@@ -28,7 +28,5 @@
         clientplaypacketlistener.onPlayerListHeader(this);
     }
 
-    public void apply(PacketListener packetlistener) {
-        this.apply((ClientPlayPacketListener) packetlistener);
-    }
+    // Loom :: decompile error
 }
