--- a/net/minecraft//world/entity/Entity.java
+++ b/net/minecraft//world/entity/Entity.java
@@ -116,6 +116,15 @@
 
 public abstract class Entity implements Nameable, EntityAccess, CommandSource {
 
+    // Loom start
+    private org.loomdev.loom.entity.EntityImpl entityImpl;
+    public org.loomdev.loom.entity.EntityImpl getLoomEntity() {
+        if (entityImpl == null) {
+            entityImpl = org.loomdev.loom.entity.LoomEntityFactory.getLoomEntity(this);
+        }
+        return entityImpl;
+    }
+    // Loom end
     protected static final Logger LOGGER = LogManager.getLogger();
     public static final String ID_TAG = "id";
     public static final String PASSENGERS_TAG = "Passengers";
@@ -159,7 +168,7 @@
     public boolean hurtMarked;
     protected Vec3 stuckSpeedMultiplier;
     @Nullable
-    private Entity.RemovalReason removalReason;
+    public Entity.RemovalReason removalReason; // Loom :: private -> public (EntityImpl)
     public static final float DEFAULT_BB_WIDTH = 0.6F;
     public static final float DEFAULT_BB_HEIGHT = 1.8F;
     public float walkDistO;
@@ -183,7 +192,7 @@
     protected Tag<Fluid> fluidOnEyes;
     public int invulnerableTime;
     protected boolean firstTick;
-    protected final SynchedEntityData entityData;
+    public final SynchedEntityData entityData; // Loom :: protected -> public
     protected static final EntityDataAccessor<Byte> DATA_SHARED_FLAGS_ID = SynchedEntityData.defineId(Entity.class, EntityDataSerializers.BYTE);
     protected static final int FLAG_ONFIRE = 0;
     private static final int FLAG_SHIFT_KEY_DOWN = 1;
@@ -203,7 +212,7 @@
     private Vec3 packetCoordinates;
     public boolean noCulling;
     public boolean hasImpulse;
-    private int portalCooldown;
+    public int portalCooldown; // Loom :: private -> public (EntityImpl)
     protected boolean isInsidePortal;
     protected int portalTime;
     protected BlockPos portalEntrancePos;
@@ -221,6 +230,7 @@
     public boolean wasOnFire;
     private float crystalSoundIntensity;
     private int lastCrystalSoundPlayTick;
+    public Optional<Boolean> fireResistantOverride = Optional.empty(); // Loom
 
     public Entity(EntityType<?> entitytype, Level level) {
         this.id = Entity.ENTITY_COUNTER.incrementAndGet();
@@ -536,7 +546,7 @@
         this.setRemainingFireTicks(0);
     }
 
-    protected void outOfWorld() {
+    public void outOfWorld() { // Loom :: protected -> public
         this.discard();
     }
 
@@ -989,7 +999,7 @@
 
     protected void onFlap() {}
 
-    protected boolean isFlapping() {
+    public boolean isFlapping() { // Loom :: protected -> public (EntityImpl)
         return false;
     }
 
@@ -1041,7 +1051,7 @@
     }
 
     public boolean fireImmune() {
-        return this.getType().fireImmune();
+        return fireResistantOverride.orElse(this.getType().fireImmune()); // Loom
     }
 
     public boolean causeFallDamage(float f, float f1, DamageSource damagesource) {
@@ -2407,7 +2417,7 @@
                     }
 
                     return PortalShape.createPortalInfo(serverlevel, blockutil_foundrectangle, direction_axis, vec3, this.getDimensions(this.getPose()), this.getDeltaMovement(), this.yRot, this.xRot);
-                }).orElse((Object) null);
+                }).orElse(null); // Loom :: decompile error
             }
         } else {
             BlockPos blockpos1;
