--- a/net/minecraft//world/entity/Entity.java
+++ b/net/minecraft//world/entity/Entity.java
@@ -117,6 +117,19 @@
 
 public abstract class Entity implements Nameable, EntityAccess, CommandSource {
 
+    // Loom start
+    public Optional<Boolean> fireResistantOverride = Optional.empty();
+    @Nullable
+    private org.loomdev.loom.entity.EntityImpl entityImpl;
+
+    public org.loomdev.loom.entity.EntityImpl getLoomEntity() {
+        if (entityImpl != null) {
+            return entityImpl;
+        }
+
+        return entityImpl = org.loomdev.loom.entity.LoomEntityFactory.getLoomEntity(this);
+    }
+    // Loom end
     protected static final Logger LOGGER = LogManager.getLogger();
     public static final String ID_TAG = "id";
     public static final String PASSENGERS_TAG = "Passengers";
@@ -160,7 +173,7 @@
     public boolean hurtMarked;
     protected Vec3 stuckSpeedMultiplier;
     @Nullable
-    private Entity.RemovalReason removalReason;
+    public Entity.RemovalReason removalReason; // Loom :: private -> public (EntityImpl)
     public static final float DEFAULT_BB_WIDTH = 0.6F;
     public static final float DEFAULT_BB_HEIGHT = 1.8F;
     public float walkDistO;
@@ -184,7 +197,7 @@
     protected Tag<Fluid> fluidOnEyes;
     public int invulnerableTime;
     protected boolean firstTick;
-    protected final SynchedEntityData entityData;
+    public final SynchedEntityData entityData; // Loom :: protected -> public
     protected static final EntityDataAccessor<Byte> DATA_SHARED_FLAGS_ID = SynchedEntityData.defineId(Entity.class, EntityDataSerializers.BYTE);
     protected static final int FLAG_ONFIRE = 0;
     private static final int FLAG_SHIFT_KEY_DOWN = 1;
@@ -204,7 +217,7 @@
     private Vec3 packetCoordinates;
     public boolean noCulling;
     public boolean hasImpulse;
-    private int portalCooldown;
+    public int portalCooldown; // Loom :: private -> public (EntityImpl)
     protected boolean isInsidePortal;
     protected int portalTime;
     protected BlockPos portalEntrancePos;
@@ -1000,7 +1013,7 @@
 
     protected void onFlap() {}
 
-    protected boolean isFlapping() {
+    public boolean isFlapping() { // Loom :: protected -> public (EntityImpl)
         return false;
     }
 
@@ -1052,7 +1065,7 @@
     }
 
     public boolean fireImmune() {
-        return this.getType().fireImmune();
+        return fireResistantOverride.orElse(this.getType().fireImmune()); // Loom
     }
 
     public boolean causeFallDamage(float f, float f1, DamageSource damagesource) {
@@ -2428,7 +2441,7 @@
                     }
 
                     return PortalShape.createPortalInfo(serverlevel, blockutil_foundrectangle, direction_axis, vec3, this.getDimensions(this.getPose()), this.getDeltaMovement(), this.getYRot(), this.getXRot());
-                }).orElse((Object) null);
+                }).orElse(null); // Loom :: decompile error
             }
         } else {
             BlockPos blockpos1;
