--- a/net/minecraft/world/entity/ExperienceOrb.java
+++ b/net/minecraft/world/entity/ExperienceOrb.java
@@ -27,11 +27,11 @@
     private static final int MAX_FOLLOW_DIST = 8;
     private static final int ORB_GROUPS_PER_AREA = 40;
     private static final double ORB_MERGE_DISTANCE = 0.5D;
-    private int age;
-    private int health;
-    private int value;
+    public int age; // Loom :: private -> public (ExperienceOrbImpl)
+    public int health; // Loom :: private -> public (ExperienceOrbImpl)
+    public int value; // Loom :: private -> public (ExperienceOrbImpl)
     private int count;
-    private Player followingPlayer;
+    public Player followingPlayer; // Loom :: private -> public (ExperienceOrbImpl)
 
     public ExperienceOrb(Level level, double d0, double d1, double d2, int i) {
         this(EntityType.EXPERIENCE_ORB, level);
