--- a/net/minecraft//world/entity/LightningBolt.java
+++ b/net/minecraft//world/entity/LightningBolt.java
@@ -39,9 +39,9 @@
     private int life;
     public long seed;
     private int flashes;
-    private boolean visualOnly;
+    public boolean visualOnly; // Loom :: private -> public (LightningBoltImpl)
     @Nullable
-    private ServerPlayer cause;
+    public ServerPlayer cause; // Loom :: private -> public (LightningBoltImpl)
     private final Set<Entity> hitEntities = Sets.newHashSet();
     private int blocksSetOnFire;
 
@@ -163,6 +163,7 @@
             BlockState blockstate = BaseFireBlock.getState(this.level, blockpos);
 
             if (this.level.getBlockState(blockpos).isAir() && blockstate.canSurvive(this.level, blockpos)) {
+                // TODO if (!org.loomdev.loom.event.LoomEventDispatcher.onBlockIgnite(level, blockpos, this, org.loomdev.api.event.block.BlockIgniteEvent.Cause.LIGHTNING).isCancelled()) // Loom :: fire BlockIgniteEvent
                 this.level.setBlockAndUpdate(blockpos, blockstate);
                 ++this.blocksSetOnFire;
             }
@@ -172,6 +173,7 @@
 
                 blockstate = BaseFireBlock.getState(this.level, blockpos1);
                 if (this.level.getBlockState(blockpos1).isAir() && blockstate.canSurvive(this.level, blockpos1)) {
+                    // TODO if (!org.loomdev.loom.event.LoomEventDispatcher.onBlockIgnite(level, blockpos1, this, org.loomdev.api.event.block.BlockIgniteEvent.Cause.LIGHTNING).isCancelled()) // Loom :: fire BlockIgniteEvent
                     this.level.setBlockAndUpdate(blockpos1, blockstate);
                     ++this.blocksSetOnFire;
                 }
@@ -237,8 +239,9 @@
             blockstate = level.getBlockState(blockpos1);
         } while (!(blockstate.getBlock() instanceof WeatheringCopper));
 
+        final BlockPos finalPos = blockpos1; // Loom - decompile error
         WeatheringCopper.getPrevious(blockstate).ifPresent((blockstate1) -> {
-            level.setBlockAndUpdate(blockpos1, blockstate1);
+            level.setBlockAndUpdate(finalPos, blockstate1); // Loom - decompile error
         });
         level.levelEvent(3002, blockpos1, -1);
         return Optional.of(blockpos1);
