--- a/net/minecraft/entity/LivingEntity.java
+++ b/net/minecraft/entity/LivingEntity.java
@@ -108,6 +108,17 @@
 
 public abstract class LivingEntity extends Entity {
 
+    // Loom start
+    public Optional<Integer> maxAirOverride = Optional.empty();
+    public Optional<Boolean> hurtByWaterOverride = Optional.empty();
+
+    @Override
+    public int getMaxAir() {
+        if (maxAirOverride == null) return super.getMaxAir(); // required because .getMaxAir() is called in the Entity class constructor.
+        return maxAirOverride.orElse(super.getMaxAir()); // Loom :: override default max air ticks
+    }
+    // Loom end
+
     private static final UUID SPRINTING_SPEED_BOOST_ID = UUID.fromString("662A6B8D-DA3E-4C1C-8813-96EA6097278D");
     private static final UUID SOUL_SPEED_BOOST_ID = UUID.fromString("87f46a96-686f-4796-b035-22e16ee9e038");
     private static final EntityAttributeModifier SPRINTING_SPEED_BOOST = new EntityAttributeModifier(LivingEntity.SPRINTING_SPEED_BOOST_ID, "Sprinting speed boost", 0.30000001192092896D, EntityAttributeModifier.Operation.MULTIPLY_TOTAL);
@@ -191,6 +202,7 @@
     private float leaningPitch;
     private float lastLeaningPitch;
     protected Brain<?> brain;
+    public boolean pickUpLoot; // Loom :: moved from MobEntity
 
     protected LivingEntity(EntityType<? extends LivingEntity> entitytype, World world) {
         super(entitytype, world);
@@ -1574,7 +1586,7 @@
 
     }
 
-    protected void destroy() {
+    public void destroy() { // Loom :: public accessor (inherited from Entity)
         this.damage(DamageSource.OUT_OF_WORLD, 4.0F);
     }
 
@@ -2363,7 +2375,7 @@
     }
 
     public boolean hurtByWater() {
-        return false;
+        return hurtByWaterOverride.orElse(false); // hurt by water override
     }
 
     private void initAi() {
@@ -2886,7 +2898,7 @@
     }
 
     public void wakeUp() {
-        Optional optional = this.getSleepingPosition();
+        Optional<BlockPos> optional = this.getSleepingPosition(); // Loom :: decompile error
         World world = this.world;
 
         this.world.getClass();
