--- a/net/minecraft/entity/vehicle/SpawnerMinecartEntity.java
+++ b/net/minecraft/entity/vehicle/SpawnerMinecartEntity.java
@@ -11,7 +11,7 @@
 
 public class SpawnerMinecartEntity extends AbstractMinecartEntity {
 
-    private final MobSpawnerLogic logic = new MobSpawnerLogic() {
+    public final MobSpawnerLogic logic = new MobSpawnerLogic() { // Loom :: public accessor
         public void sendStatus(World world, BlockPos blockpos, int i) {
             world.sendEntityStatus(SpawnerMinecartEntity.this, (byte) i);
         }
