--- a/net/minecraft/server/network/ServerPlayerInteractionManager.java
+++ b/net/minecraft/server/network/ServerPlayerInteractionManager.java
@@ -101,7 +101,11 @@
 
                 if (f >= 1.0F) {
                     this.failedToMine = false;
+                    // Loom start :: fire BlockBrokenEvent
+                    if (!org.loomdev.loom.event.LoomEventDispatcher.onBlockBroken(this.world, this.failedMiningPos, this.player).isCancelled()) {
                     this.tryBreakBlock(this.failedMiningPos);
+                    }
+                    // Loom end
                 }
             }
         } else if (this.mining) {
@@ -229,7 +233,6 @@
         } else {
             this.player.networkHandler.sendPacket(new PlayerActionResponseS2CPacket(blockpos, this.world.getBlockState(blockpos), playeractionc2spacket_action, false, s));
         }
-
     }
 
     public boolean tryBreakBlock(BlockPos blockpos) {
@@ -240,7 +243,11 @@
         } else {
             BlockEntity blockentity = this.world.getBlockEntity(blockpos);
             Block block = blockstate.getBlock();
-
+            // Loom start :: fire BlockBrokenEvent
+            if (org.loomdev.loom.event.LoomEventDispatcher.onBlockBroken(world, blockpos, player).isCancelled()) {
+                return false;
+            }
+            // Loom end
             if ((block instanceof CommandBlock || block instanceof StructureBlock || block instanceof JigsawBlock) && !this.player.isCreativeLevelTwoOp()) {
                 this.world.updateListeners(blockpos, blockstate, blockstate, 3);
                 return false;
