--- a/net/minecraft/server/network/ServerPlayerEntity.java
+++ b/net/minecraft/server/network/ServerPlayerEntity.java
@@ -133,6 +133,8 @@
 
 public class ServerPlayerEntity extends PlayerEntity implements ScreenHandlerListener {
 
+    public java.util.OptionalInt viewDistance = java.util.OptionalInt.empty(); // Loom :: expose client view distance
+
     private static final Logger LOGGER = LogManager.getLogger();
     public ServerPlayNetworkHandler networkHandler;
     public final MinecraftServer server;
@@ -317,7 +319,7 @@
 
             logger.getClass();
             dataresult.resultOrPartial(logger::error).ifPresent((tag) -> {
-                compoundtag.put("SpawnDimension", tag);
+                compoundtag.put("SpawnDimension", (net.minecraft.nbt.Tag) tag); // Loom :: decompile error
             });
         }
 
@@ -1252,6 +1254,7 @@
         this.clientChatColorsEnabled = clientsettingsc2spacket.hasChatColors();
         this.getDataTracker().set(ServerPlayerEntity.PLAYER_MODEL_PARTS, (byte) clientsettingsc2spacket.getPlayerModelBitMask());
         this.getDataTracker().set(ServerPlayerEntity.MAIN_ARM, (byte) (clientsettingsc2spacket.getMainArm() == Arm.LEFT ? 0 : 1));
+        this.viewDistance = java.util.OptionalInt.of(clientsettingsc2spacket.viewDistance);
     }
 
     public ChatVisibility getClientChatVisibility() {
@@ -1288,7 +1291,7 @@
     }
 
     public void onStartedTracking(Entity entity) {
-        this.removedEntities.remove(entity.getEntityId());
+        this.removedEntities.remove(Integer.valueOf(entity.getEntityId())); // Loom :: decompile error
     }
 
     protected void updatePotionVisibility() {
