--- a/net/minecraft/server/network/ServerQueryNetworkHandler.java
+++ b/net/minecraft/server/network/ServerQueryNetworkHandler.java
@@ -33,7 +33,19 @@
             this.connection.disconnect(ServerQueryNetworkHandler.REQUEST_HANDLED);
         } else {
             this.responseSent = true;
+            // Loom start :: fire ServerPingedEvent
+            org.loomdev.loom.event.LoomEventDispatcher.onServerPing(this.connection, this.server.getServerMetadata()).thenAccept(event -> {
+                if (event.isCancelled()) {
+                    return;
+                }
+
+                net.minecraft.server.ServerMetadata metadata = this.server.getServerMetadata();
+                metadata.setDescription(org.loomdev.loom.util.transformer.TextTransformer.toMinecraft(event.getMotd()));
+                metadata.setPlayers(new net.minecraft.server.ServerMetadata.Players(event.getSlots(), event.getOnlinePlayers()));
+                metadata.setFavicon(event.getFavicon());
             this.connection.send(new QueryResponseS2CPacket(this.server.getServerMetadata()));
+            });
+            // Loom end
         }
     }
 
