--- a/net/minecraft//world/level/block/CampfireBlock.java
+++ b/net/minecraft//world/level/block/CampfireBlock.java
@@ -196,6 +196,7 @@
             if (flag && !(Boolean) blockstate.getValue(CampfireBlock.LIT) && !(Boolean) blockstate.getValue(CampfireBlock.WATERLOGGED)) {
                 BlockPos blockpos = blockhitresult.getBlockPos();
 
+                // TODO if (org.loomdev.loom.event.LoomEventDispatcher.onBlockIgnite(level, blockpos, projectile, projectile instanceof net.minecraft.world.entity.projectile.Fireball ? org.loomdev.api.event.block.BlockIgniteEvent.Cause.FIREBALL : org.loomdev.api.event.block.BlockIgniteEvent.Cause.FLAME_ARROW).isCancelled()) return; // Loom :: fire BlockIgniteEvent
                 level.setBlock(blockpos, (BlockState) blockstate.setValue(BlockStateProperties.LIT, true), 11);
             }
         }
