--- a/net/minecraft/world/level/block/ChangeOverTimeBlock.java
+++ b/net/minecraft/world/level/block/ChangeOverTimeBlock.java
@@ -67,7 +67,11 @@
 
         if (random.nextFloat() < f1) {
             this.getNext(blockstate).ifPresent((blockstate2) -> {
-                serverlevel.setBlockAndUpdate(blockpos, blockstate2);
+                // Loom start :: fire BlockEvent.ChangeOverTime
+                var event = org.loomdev.loom.event.LoomEventDispatcher.fire(new org.loomdev.loom.event.block.BlockEventImpl.ChangeOverTimeImpl(serverlevel, blockpos, blockstate2));
+                if (event.isCanceled()) return;
+                serverlevel.setBlockAndUpdate(blockpos, event.getMinecraftState());
+                // Loom end
             });
         }
 
