--- a/net/minecraft/world/level/block/ChangeOverTimeBlock.java
+++ b/net/minecraft/world/level/block/ChangeOverTimeBlock.java
@@ -18,6 +18,10 @@
     }
 
     default void change(Level level, BlockState blockstate, BlockPos blockpos) {
-        level.setBlockAndUpdate(blockpos, this.getChangeTo(blockstate));
+        // Loom start :: fire BlockEvent.ChangeOverTime
+        var event = org.loomdev.loom.event.LoomEventDispatcher.onBlockChangeOverTime(level, blockpos, getChangeTo(blockstate));
+        if (event.isCancelable()) return;
+        level.setBlockAndUpdate(blockpos, this.getChangeTo(((org.loomdev.loom.block.BlockStateImpl) event.getNewState()).getMinecraftState()));
+        // Loom end
     }
 }
