--- a/net/minecraft/world/level/block/ChangeOverTimeBlock.java
+++ b/net/minecraft/world/level/block/ChangeOverTimeBlock.java
@@ -65,7 +65,11 @@
         float f1 = f * f * this.getChanceModifier();
 
         if (random.nextFloat() < f1) {
-            serverlevel.setBlockAndUpdate(blockpos, this.getChangeTo(blockstate));
+            // Loom start :: fire BlockEvent.ChangeOverTime
+            var event = org.loomdev.loom.event.LoomEventDispatcher.fire(new org.loomdev.loom.event.block.BlockEventImpl.ChangeOverTimeImpl(serverlevel, blockpos, getChangeTo(blockstate)));
+            if (event.isCanceled()) return;
+            serverlevel.setBlockAndUpdate(blockpos, event.getMinecraftState());
+            // Loom end
         }
 
     }
