--- a/net/minecraft//world/level/block/CoralPlantBlock.java
+++ b/net/minecraft//world/level/block/CoralPlantBlock.java
@@ -30,7 +30,12 @@
 
     public void tick(BlockState blockstate, ServerLevel serverlevel, BlockPos blockpos, Random random) {
         if (!scanForWater(blockstate, serverlevel, blockpos)) {
-            serverlevel.setBlock(blockpos, (BlockState) this.deadBlock.defaultBlockState().setValue(CoralPlantBlock.WATERLOGGED, false), 2);
+            // Loom start :: fire PlantEvent.Die
+            var newState = this.deadBlock.defaultBlockState().setValue(CoralPlantBlock.WATERLOGGED, false);
+            var event = org.loomdev.loom.event.LoomEventDispatcher.fire(new org.loomdev.loom.event.block.PlantEventImpl.DieImpl(serverlevel, blockpos, newState));
+            if (event.isCanceled()) return;
+            serverlevel.setBlock(blockpos, event.getMinecraftState(), 2);
+            // Loom end
         }
 
     }
