--- a/net/minecraft/world/level/block/CoralBlock.java
+++ b/net/minecraft/world/level/block/CoralBlock.java
@@ -24,7 +24,11 @@
 
     public void tick(BlockState blockstate, ServerLevel serverlevel, BlockPos blockpos, Random random) {
         if (!this.scanForWater(serverlevel, blockpos)) {
-            serverlevel.setBlock(blockpos, this.deadBlock.defaultBlockState(), 2);
+            // Loom start :: fire PlantEvent.Die
+            var event = org.loomdev.loom.event.LoomEventDispatcher.fire(new org.loomdev.loom.event.block.PlantEventImpl.DieImpl(serverlevel, blockpos, deadBlock.defaultBlockState()));
+            if (event.isCanceled()) return;
+            serverlevel.setBlock(blockpos, event.getMinecraftState(), 2);
+            // Loom end
         }
 
     }
