--- a/net/minecraft/world/level/block/CoralBlock.java
+++ b/net/minecraft/world/level/block/CoralBlock.java
@@ -24,7 +24,11 @@
 
     public void tick(BlockState blockstate, ServerLevel serverlevel, BlockPos blockpos, Random random) {
         if (!this.scanForWater(serverlevel, blockpos)) {
-            serverlevel.setBlock(blockpos, this.deadBlock.defaultBlockState(), 2);
+            // Loom start :: fire PlantEvent.Die
+            var event = org.loomdev.loom.event.LoomEventDispatcher.onPlantDie(serverlevel, blockpos, deadBlock.defaultBlockState());
+            if (event.isCanceled()) return;
+            serverlevel.setBlock(blockpos, ((org.loomdev.loom.block.BlockStateImpl) event.getNewState()).getMinecraftState(), 2);
+            // Loom end
         }
 
     }
