--- a/net/minecraft//world/level/block/LeavesBlock.java
+++ b/net/minecraft//world/level/block/LeavesBlock.java
@@ -42,6 +42,7 @@
 
     public void randomTick(BlockState blockstate, ServerLevel serverlevel, BlockPos blockpos, Random random) {
         if (!(Boolean) blockstate.getValue(LeavesBlock.PERSISTENT) && (Integer) blockstate.getValue(LeavesBlock.DISTANCE) == 7) {
+            if (org.loomdev.loom.event.LoomEventDispatcher.fire(new org.loomdev.loom.event.block.PlantEventImpl.DecayImpl(serverlevel, blockpos)).isCanceled()) return; // Loom :: fire PlantEvent.Decay
             dropResources(blockstate, serverlevel, blockpos);
             serverlevel.removeBlock(blockpos, false);
         }
