--- a/net/minecraft/world/level/block/SpongeBlock.java
+++ b/net/minecraft/world/level/block/SpongeBlock.java
@@ -32,6 +32,7 @@
     }
 
     protected void tryAbsorbWater(Level level, BlockPos blockpos) {
+        if (false) // Loom start :: move down block updates
         if (this.removeWaterBreadthFirstSearch(level, blockpos)) {
             level.setBlock(blockpos, Blocks.WET_SPONGE.defaultBlockState(), 2);
             level.levelEvent(2001, blockpos, Block.getId(Blocks.WATER.defaultBlockState()));
@@ -41,6 +42,7 @@
 
     private boolean removeWaterBreadthFirstSearch(Level level, BlockPos blockpos) {
         Queue<Tuple<BlockPos, Integer>> queue = Lists.newLinkedList();
+        java.util.Set<org.loomdev.api.block.Block> absorbedBlocks = new java.util.HashSet<>(); // Loom
 
         queue.add(new Tuple<>(blockpos, 0));
         int i = 0;
@@ -60,27 +62,32 @@
                 Material material = blockstate.getMaterial();
 
                 if (fluidstate.is(FluidTags.WATER)) {
-                    if (blockstate.getBlock() instanceof BucketPickup && ((BucketPickup) blockstate.getBlock()).takeLiquid(level, blockpos2, blockstate) != Fluids.EMPTY) {
+                    // Loom start
+                    if (blockstate.getBlock() instanceof BucketPickup /* && ((BucketPickup) blockstate.getBlock()).takeLiquid(level, blockpos2, blockstate) != Fluids.EMPTY */) {
+                        absorbedBlocks.add(org.loomdev.loom.block.BlockImpl.at(level, blockpos2));
                         ++i;
                         if (j < 6) {
                             queue.add(new Tuple<>(blockpos2, j + 1));
                         }
                     } else if (blockstate.getBlock() instanceof LiquidBlock) {
-                        level.setBlock(blockpos2, Blocks.AIR.defaultBlockState(), 3);
+                        // level.setBlock(blockpos2, Blocks.AIR.defaultBlockState(), 3);
+                        absorbedBlocks.add(org.loomdev.loom.block.BlockImpl.at(level, blockpos2));
                         ++i;
                         if (j < 6) {
                             queue.add(new Tuple<>(blockpos2, j + 1));
                         }
                     } else if (material == Material.WATER_PLANT || material == Material.REPLACEABLE_WATER_PLANT) {
-                        BlockEntity blockentity = blockstate.hasBlockEntity() ? level.getBlockEntity(blockpos2) : null;
+                        // BlockEntity blockentity = blockstate.hasBlockEntity() ? level.getBlockEntity(blockpos2) : null;
 
-                        dropResources(blockstate, level, blockpos2, blockentity);
-                        level.setBlock(blockpos2, Blocks.AIR.defaultBlockState(), 3);
+                        // dropResources(blockstate, level, blockpos2, blockentity);
+                        // level.setBlock(blockpos2, Blocks.AIR.defaultBlockState(), 3);
+                        absorbedBlocks.add(org.loomdev.loom.block.BlockImpl.at(level, blockpos2));
                         ++i;
                         if (j < 6) {
                             queue.add(new Tuple<>(blockpos2, j + 1));
                         }
                     }
+                    // Loom end
                 }
             }
 
@@ -89,6 +96,35 @@
             }
         }
 
+        // Loom start :: fire BlockAbsorbEvent
+        final int finali = i;
+        var event = org.loomdev.loom.event.LoomEventDispatcher.onBlockAbsorb(level, blockpos, absorbedBlocks);
+
+        if (!event.isCancelled()) {
+            for (var block : event.getAbsorbedBlocks()) {
+                var blockImpl = (org.loomdev.loom.block.BlockImpl) block;
+                var state = blockImpl.getMinecraftBlockState();
+                var pos = blockImpl.getMinecraftPos();
+                var level1 = (Level) blockImpl.getMinecraftWorld();
+                var material1 = state.getMaterial();
+
+                if (state.getBlock() instanceof BucketPickup && ((BucketPickup) blockImpl.getMinecraftBlockState().getBlock()).takeLiquid(level1, pos, state) != Fluids.EMPTY) {
+                    // NOOP
+                } else if (state.getBlock() instanceof LiquidBlock) {
+                    level1.setBlock(pos, Blocks.AIR.defaultBlockState(), 3);
+                } else if (material1 == Material.WATER_PLANT || material1 == Material.REPLACEABLE_WATER_PLANT) {
+                    BlockEntity blockentity = state.hasBlockEntity() ? level1.getBlockEntity(pos) : null;
+                    dropResources(state, level1, pos, blockentity);
+                    level1.setBlock(pos, Blocks.AIR.defaultBlockState(), 3);
+                }
+            }
+
+            if (finali > 0) {
+                level.setBlock(blockpos, Blocks.WET_SPONGE.defaultBlockState(), 2);
+                level.levelEvent(2001, blockpos, Block.getId(Blocks.WATER.defaultBlockState()));
+            }
+        }
+        // Loom end
         return i > 0;
     }
 }
