--- a/net/minecraft//world/level/block/SpongeBlock.java
+++ b/net/minecraft//world/level/block/SpongeBlock.java
@@ -35,6 +35,7 @@
 
     protected void tryAbsorbWater(Level level, BlockPos blockpos) {
         if (this.removeWaterBreadthFirstSearch(level, blockpos)) {
+            if (true) return; // Loom :: move down block updates
             level.setBlock(blockpos, Blocks.WET_SPONGE.defaultBlockState(), 2);
             level.levelEvent(2001, blockpos, Block.getId(Blocks.WATER.defaultBlockState()));
         }
@@ -43,6 +44,7 @@
 
     private boolean removeWaterBreadthFirstSearch(Level level, BlockPos blockpos) {
         Queue<Tuple<BlockPos, Integer>> queue = Lists.newLinkedList();
+        java.util.List<org.loomdev.api.block.BlockPointer> absorbedBlocks = new java.util.ArrayList<>(); // Loom
 
         queue.add(new Tuple<>(blockpos, 0));
         int i = 0;
@@ -62,22 +64,27 @@
                 Material material = blockstate.getMaterial();
 
                 if (fluidstate.is(FluidTags.WATER)) {
-                    if (blockstate.getBlock() instanceof BucketPickup && !((BucketPickup) blockstate.getBlock()).pickupBlock(level, blockpos2, blockstate).isEmpty()) {
+                    // Loom start
+                    if (blockstate.getBlock() instanceof BucketPickup /* && ((BucketPickup) blockstate.getBlock()).pickupBlock(level, blockpos2, blockstate).isEmpty() */) {
+                        absorbedBlocks.add(new org.loomdev.loom.block.BlockPointerImpl(level, blockpos2));
                         ++i;
                         if (j < 6) {
                             queue.add(new Tuple<>(blockpos2, j + 1));
                         }
                     } else if (blockstate.getBlock() instanceof LiquidBlock) {
-                        level.setBlock(blockpos2, Blocks.AIR.defaultBlockState(), 3);
+                        // level.setBlock(blockpos2, Blocks.AIR.defaultBlockState(), 3);
+                        absorbedBlocks.add(new org.loomdev.loom.block.BlockPointerImpl(level, blockpos2));
                         ++i;
                         if (j < 6) {
                             queue.add(new Tuple<>(blockpos2, j + 1));
                         }
                     } else if (material == Material.WATER_PLANT || material == Material.REPLACEABLE_WATER_PLANT) {
-                        BlockEntity blockentity = blockstate.hasBlockEntity() ? level.getBlockEntity(blockpos2) : null;
+                        // BlockEntity blockentity = blockstate.hasBlockEntity() ? level.getBlockEntity(blockpos2) : null;
 
-                        dropResources(blockstate, level, blockpos2, blockentity);
-                        level.setBlock(blockpos2, Blocks.AIR.defaultBlockState(), 3);
+                        // dropResources(blockstate, level, blockpos2, blockentity);
+                        // level.setBlock(blockpos2, Blocks.AIR.defaultBlockState(), 3);
+                        absorbedBlocks.add(new org.loomdev.loom.block.BlockPointerImpl(level, blockpos2));
+                        // Loom end
                         ++i;
                         if (j < 6) {
                             queue.add(new Tuple<>(blockpos2, j + 1));
@@ -91,6 +98,35 @@
             }
         }
 
+        // Loom start :: fire SpongeEvent.Absorb
+        final int finali = i;
+        var event = org.loomdev.loom.event.LoomEventDispatcher.fire(new org.loomdev.loom.event.block.SpongeEventImpl.AbsorbImpl(level, blockpos, absorbedBlocks));
+
+        if (!event.isCanceled()) {
+            for (var block : event.getAbsorbedBlocks()) {
+                var pointer = (org.loomdev.loom.block.BlockPointerImpl) block;
+                var state = ((org.loomdev.loom.block.BlockStateImpl) pointer.getBlockState()).getMinecraftState();
+                var pos = pointer.getMinecraftPos();
+                var level1 = pointer.getMinecraftWorld();
+                var material1 = state.getMaterial();
+
+                if (state.getBlock() instanceof BucketPickup && ((BucketPickup) state.getBlock()).pickupBlock(level1, pos, state).isEmpty()) {
+                    // NOOP
+                } else if (state.getBlock() instanceof LiquidBlock) {
+                    level1.setBlock(pos, Blocks.AIR.defaultBlockState(), 3);
+                } else if (material1 == Material.WATER_PLANT || material1 == Material.REPLACEABLE_WATER_PLANT) {
+                    BlockEntity blockentity = state.hasBlockEntity() ? level1.getBlockEntity(pos) : null;
+                    dropResources(state, level1, pos, blockentity);
+                    level1.setBlock(pos, Blocks.AIR.defaultBlockState(), 3);
+                }
+            }
+
+            if (finali > 0) {
+                level.setBlock(blockpos, Blocks.WET_SPONGE.defaultBlockState(), 2);
+                level.levelEvent(2001, blockpos, Block.getId(Blocks.WATER.defaultBlockState()));
+            }
+        }
+        // Loom end
         return i > 0;
     }
 }
