--- a/net/minecraft/entity/mob/BlazeEntity.java
+++ b/net/minecraft/entity/mob/BlazeEntity.java
@@ -96,7 +96,7 @@
     }
 
     public boolean hurtByWater() {
-        return true;
+        return hurtByWaterOverride.orElse(true); // hurt by water override
     }
 
     protected void mobTick() {
@@ -126,11 +126,11 @@
         return this.isFireActive();
     }
 
-    private boolean isFireActive() {
+    public boolean isFireActive() { // Loom :: public accessor
         return ((Byte) this.dataTracker.get(BlazeEntity.BLAZE_FLAGS) & 1) != 0;
     }
 
-    private void setFireActive(boolean flag) {
+    public void setFireActive(boolean flag) { // Loom :: public accessor
         byte b0 = (Byte) this.dataTracker.get(BlazeEntity.BLAZE_FLAGS);
 
         if (flag) {
