--- a/net/minecraft/entity/mob/CreeperEntity.java
+++ b/net/minecraft/entity/mob/CreeperEntity.java
@@ -43,9 +43,9 @@
     private static final TrackedData<Boolean> CHARGED = DataTracker.registerData(CreeperEntity.class, TrackedDataHandlerRegistry.BOOLEAN);
     private static final TrackedData<Boolean> IGNITED = DataTracker.registerData(CreeperEntity.class, TrackedDataHandlerRegistry.BOOLEAN);
     private int lastFuseTime;
-    private int currentFuseTime;
-    private int fuseTime = 30;
-    private int explosionRadius = 3;
+    public int currentFuseTime; // Loom :: public access
+    public int fuseTime = 30; // Loom :: public access
+    public int explosionRadius = 3; // Loom :: public access
     private int headsDropped;
 
     public CreeperEntity(EntityType<? extends CreeperEntity> entitytype, World world) {
@@ -187,8 +187,20 @@
 
     public void onStruckByLightning(LightningEntity lightningentity) {
         super.onStruckByLightning(lightningentity);
+        // Loom start :: call CreeperChargedEvent
+        org.loomdev.loom.event.LoomEventDispatcher.onCreeperCharged(this, lightningentity).thenAccept(event -> {
+            if (!event.isCancelled()) {
         this.dataTracker.set(CreeperEntity.CHARGED, true);
+            }
+        });
+        // Loom end
+    }
+
+    // Loom start :: set charged method
+    public void setCharged(boolean charged) {
+        this.dataTracker.set(CreeperEntity.CHARGED, charged);
     }
+    // Loom end
 
     protected ActionResult interactMob(PlayerEntity playerentity, Hand hand) {
         ItemStack itemstack = playerentity.getStackInHand(hand);
@@ -208,7 +220,7 @@
         }
     }
 
-    private void explode() {
+    public void explode() { // Loom :: public access
         if (!this.world.isClient) {
             Explosion.DestructionType explosion_destructiontype = this.world.getGameRules().getBoolean(GameRules.DO_MOB_GRIEFING) ? Explosion.DestructionType.DESTROY : Explosion.DestructionType.NONE;
             float f = this.T_() ? 2.0F : 1.0F;
@@ -250,7 +262,13 @@
     }
 
     public void setIgnited() {
+        // Loom start :: fire CreeperIgnitedEvent
+        org.loomdev.loom.event.LoomEventDispatcher.onCreeperIgnited(this).thenAccept(event -> {
+            if (!event.isCancelled()) {
         this.dataTracker.set(CreeperEntity.IGNITED, true);
+            }
+        });
+        // Loom end
     }
 
     public boolean shouldDropHead() {
