--- a/net/minecraft/entity/mob/CreeperEntity.java
+++ b/net/minecraft/entity/mob/CreeperEntity.java
@@ -40,13 +40,19 @@
 
 public class CreeperEntity extends HostileEntity {
 
+    // Loom start :: additions
+    public void setCharged(boolean charged) {
+        this.dataTracker.set(CreeperEntity.CHARGED, charged);
+    }
+    // Loom end :: additions
+
     private static final TrackedData<Integer> FUSE_SPEED = DataTracker.registerData(CreeperEntity.class, TrackedDataHandlerRegistry.INTEGER);
     private static final TrackedData<Boolean> CHARGED = DataTracker.registerData(CreeperEntity.class, TrackedDataHandlerRegistry.BOOLEAN);
     private static final TrackedData<Boolean> IGNITED = DataTracker.registerData(CreeperEntity.class, TrackedDataHandlerRegistry.BOOLEAN);
     private int lastFuseTime;
-    private int currentFuseTime;
-    private int fuseTime = 30;
-    private int explosionRadius = 3;
+    public int currentFuseTime; // Loom :: public access
+    public int fuseTime = 30; // Loom :: public access
+    public int explosionRadius = 3; // Loom :: public access
     private int headsDropped;
 
     public CreeperEntity(EntityType<? extends CreeperEntity> entitytype, World world) {
@@ -209,7 +215,7 @@
         }
     }
 
-    private void explode() {
+    public void explode() { // Loom :: public access
         if (!this.world.isClient) {
             Explosion.DestructionType explosion_destructiontype = this.world.getGameRules().getBoolean(GameRules.DO_MOB_GRIEFING) ? Explosion.DestructionType.DESTROY : Explosion.DestructionType.NONE;
             float f = this.S_() ? 2.0F : 1.0F;
@@ -251,7 +257,13 @@
     }
 
     public void ignite() {
+        // Loom start :: fire CreeperIgnitedEvent
+        org.loomdev.loom.event.LoomEventDispatcher.onCreeperIgnited(this).thenAccept(event -> {
+            if (!event.isCancelled()) {
         this.dataTracker.set(CreeperEntity.IGNITED, true);
+            }
+        });
+        // Loom end
     }
 
     public boolean shouldDropHead() {
