--- a/net/minecraft/entity/mob/HostileEntity.java
+++ b/net/minecraft/entity/mob/HostileEntity.java
@@ -23,6 +23,13 @@
 
 public abstract class HostileEntity extends PathAwareEntity implements Monster {
 
+    // Loom start :: getLoomEntity
+    @Override
+    public org.loomdev.loom.entity.mob.HostileEntityImpl getLoomEntity() {
+        return (org.loomdev.loom.entity.mob.HostileEntityImpl) super.getLoomEntity();
+    }
+    // Loom end :: getLoomEntity
+
     protected HostileEntity(EntityType<? extends HostileEntity> entitytype, World world) {
         super(entitytype, world);
         this.experiencePoints = 5;
