--- a/net/minecraft/entity/mob/MobEntity.java
+++ b/net/minecraft/entity/mob/MobEntity.java
@@ -95,8 +95,8 @@
     protected final float[] handDropChances;
     private final DefaultedList<ItemStack> armorItems;
     protected final float[] armorDropChances;
-    private boolean pickUpLoot;
-    private boolean persistent;
+    // private boolean pickUpLoot; // Loom :: moved to LivingEntity
+    public boolean persistent; // Loom :: public accessor
     private final Map<PathNodeType, Float> pathfindingPenalties;
     private Identifier lootTable;
     private long lootTableSeed;
@@ -1150,7 +1150,7 @@
         if (this.removed) {
             return null;
         } else {
-            T t0 = (MobEntity) entitytype.create(this.world);
+            T t0 = (T) entitytype.create(this.world); // Loom :: decompile error
 
             t0.copyPositionAndRotation(this);
             t0.setBaby(this.isBaby());
