--- a/net/minecraft/entity/mob/PatrolEntity.java
+++ b/net/minecraft/entity/mob/PatrolEntity.java
@@ -27,7 +27,7 @@
 
     private BlockPos patrolTarget;
     private boolean patrolLeader;
-    private boolean patrolling;
+    public boolean patrolling; // Loom :: public accessor
 
     protected PatrolEntity(EntityType<? extends PatrolEntity> entitytype, World world) {
         super(entitytype, world);
