--- a/net/minecraft/entity/mob/SpellcastingIllagerEntity.java
+++ b/net/minecraft/entity/mob/SpellcastingIllagerEntity.java
@@ -18,7 +18,7 @@
 
     private static final TrackedData<Byte> SPELL = DataTracker.registerData(SpellcastingIllagerEntity.class, TrackedDataHandlerRegistry.BYTE);
     protected int spellTicks;
-    private SpellcastingIllagerEntity.Spell spell;
+    public SpellcastingIllagerEntity.Spell spell; // Loom :: public accessor
 
     protected SpellcastingIllagerEntity(EntityType<? extends SpellcastingIllagerEntity> entitytype, World world) {
         super(entitytype, world);
@@ -88,7 +88,7 @@
 
         NONE(0, 0.0D, 0.0D, 0.0D), SUMMON_VEX(1, 0.7D, 0.7D, 0.8D), FANGS(2, 0.4D, 0.3D, 0.35D), WOLOLO(3, 0.7D, 0.5D, 0.2D), DISAPPEAR(4, 0.3D, 0.3D, 0.8D), BLINDNESS(5, 0.1D, 0.1D, 0.2D);
 
-        private final int id;
+        public final int id; // Loom :: public accessor
         private final double[] particleVelocity;
 
         private Spell(int i, double d0, double d1, double d2) {
