--- a/net/minecraft/entity/projectile/FireworkRocketEntity.java
+++ b/net/minecraft/entity/projectile/FireworkRocketEntity.java
@@ -31,7 +31,7 @@
 
     private static final TrackedData<ItemStack> ITEM = DataTracker.registerData(FireworkRocketEntity.class, TrackedDataHandlerRegistry.ITEM_STACK);
     private static final TrackedData<OptionalInt> SHOOTER_ENTITY_ID = DataTracker.registerData(FireworkRocketEntity.class, TrackedDataHandlerRegistry.FIREWORK_DATA);
-    private static final TrackedData<Boolean> SHOT_AT_ANGLE = DataTracker.registerData(FireworkRocketEntity.class, TrackedDataHandlerRegistry.BOOLEAN);
+    public static final TrackedData<Boolean> SHOT_AT_ANGLE = DataTracker.registerData(FireworkRocketEntity.class, TrackedDataHandlerRegistry.BOOLEAN); // Loom :: public accessor
     private int life;
     private int lifeTime;
     private LivingEntity shooter;
@@ -146,7 +146,7 @@
 
     }
 
-    private void explodeAndRemove() {
+    public void explodeAndRemove() { // Loom :: public accessor
         this.world.sendEntityStatus(this, (byte) 17);
         this.explode();
         this.method_31472();
