--- a/net/minecraft/entity/projectile/ProjectileEntity.java
+++ b/net/minecraft/entity/projectile/ProjectileEntity.java
@@ -23,7 +23,7 @@
     private int ownerEntityId;
     private boolean leftOwner;
 
-    ProjectileEntity(EntityType<? extends ProjectileEntity> entitytype, World world) {
+    public ProjectileEntity(EntityType<? extends ProjectileEntity> entitytype, World world) {
         super(entitytype, world);
     }
 
@@ -147,7 +147,7 @@
         this.yaw = updateRotation(this.prevYaw, (float) (MathHelper.atan2(vec3d.x, vec3d.z) * 57.2957763671875D));
     }
 
-    protected static float updateRotation(float f, float f1) {
+    public static float updateRotation(float f, float f1) { // Loom :: decompile error?
         while (f1 - f < -180.0F) {
             f -= 360.0F;
         }
