--- a/net/minecraft/entity/projectile/ShulkerBulletEntity.java
+++ b/net/minecraft/entity/projectile/ShulkerBulletEntity.java
@@ -29,7 +29,7 @@
 
 public class ShulkerBulletEntity extends ProjectileEntity {
 
-    private Entity target;
+    public Entity target; // Loom :: public accessor
     @Nullable
     private Direction direction;
     private int stepCount;
@@ -37,7 +37,7 @@
     private double targetY;
     private double targetZ;
     @Nullable
-    private UUID targetUuid;
+    public UUID targetUuid; // Loom :: public accessor
 
     public ShulkerBulletEntity(EntityType<? extends ShulkerBulletEntity> entitytype, World world) {
         super(entitytype, world);
