--- a/net/minecraft/world/poi/PointOfInterestType.java
+++ b/net/minecraft/world/poi/PointOfInterestType.java
@@ -61,7 +61,7 @@
     public static final PointOfInterestType LODESTONE = register("lodestone", getAllStatesOf(Blocks.LODESTONE), 0, 1);
     protected static final Set<BlockState> REGISTERED_STATES = new ObjectOpenHashSet(PointOfInterestType.BLOCK_STATE_TO_POINT_OF_INTEREST_TYPE.keySet());
     private final String id;
-    private final Set<BlockState> blockStates;
+    public final Set<BlockState> blockStates; // Loom :: public accessor
     private final int ticketCount;
     private final Predicate<PointOfInterestType> completionCondition;
     private final int searchDistance;
