--- a/net/minecraft//world/entity/vehicle/Boat.java
+++ b/net/minecraft//world/entity/vehicle/Boat.java
@@ -413,7 +413,7 @@
         return this.getPaddleState(i) ? (float) Mth.clampedLerp((double) this.paddlePositions[i] - 0.39269909262657166D, (double) this.paddlePositions[i], (double) f) : 0.0F;
     }
 
-    private Boat.Status getStatus() {
+    public Boat.Status getStatus() { // Loom :: private -> public (BoatImpl)
         Boat.Status boat_status = this.isUnderwater();
 
         if (boat_status != null) {
