--- a/net/minecraft/entity/boss/dragon/EnderDragonFight.java
+++ b/net/minecraft/entity/boss/dragon/EnderDragonFight.java
@@ -59,7 +59,7 @@
 
     private static final Logger LOGGER = LogManager.getLogger();
     private static final Predicate<Entity> VALID_ENTITY = EntityPredicates.VALID_ENTITY.and(EntityPredicates.maxDistance(0.0D, 128.0D, 0.0D, 192.0D));
-    private final ServerBossBar bossBar;
+    public final ServerBossBar bossBar; // Loom :: public accessor
     private final ServerWorld world;
     private final List<Integer> gateways;
     private final BlockPattern endPortalPattern;
@@ -71,8 +71,8 @@
     private boolean previouslyKilled;
     private UUID dragonUuid;
     private boolean doLegacyCheck;
-    private BlockPos exitPortalLocation;
-    private EnderDragonSpawnState dragonSpawnState;
+    public BlockPos exitPortalLocation; // Loom :: public accessor
+    public EnderDragonSpawnState dragonSpawnState; // Loom :: public accessor
     private int spawnStateTimer;
     private List<EndCrystalEntity> crystals;
 
@@ -233,7 +233,7 @@
 
     }
 
-    protected void setSpawnState(EnderDragonSpawnState enderdragonspawnstate) {
+    public void setSpawnState(EnderDragonSpawnState enderdragonspawnstate) { // Loom :: public accessor
         if (this.dragonSpawnState == null) {
             throw new IllegalStateException("Dragon respawn isn't in progress, can't skip ahead in the animation.");
         } else {
@@ -418,7 +418,7 @@
         ConfiguredFeatures.END_GATEWAY_DELAYED.generate(this.world, this.world.getChunkManager().getChunkGenerator(), new Random(), blockpos);
     }
 
-    private void generateEndPortal(boolean flag) {
+    public void generateEndPortal(boolean flag) { // Loom :: public accessor
         EndPortalFeature endportalfeature = new EndPortalFeature(flag);
 
         if (this.exitPortalLocation == null) {
