--- a/net/minecraft/fluid/FlowableFluid.java
+++ b/net/minecraft/fluid/FlowableFluid.java
@@ -433,6 +433,12 @@
             FluidState fluidstate1 = this.getUpdatedState(world, blockpos, world.getBlockState(blockpos));
             int i = this.getNextTickDelay(world, blockpos, fluidstate, fluidstate1);
 
+            // Loom start :: fire FluidLevelChangedEvent
+            if (org.loomdev.loom.event.LoomEventDispatcher.onFluidLevelChange(world, blockpos).isCancelled()) {
+                return;
+            }
+            // Loom end
+
             if (fluidstate1.isEmpty()) {
                 fluidstate = fluidstate1;
                 world.setBlockState(blockpos, Blocks.AIR.getDefaultState(), 3);
