--- a/net/minecraft/fluid/LavaFluid.java
+++ b/net/minecraft/fluid/LavaFluid.java
@@ -50,6 +50,11 @@
 
                     if (blockstate.isAir()) {
                         if (this.canLightFire(world, blockpos1)) {
+                            // Loom start :: fire BlockIgnitedEvent
+                            if (org.loomdev.loom.event.LoomEventDispatcher.onBlockIgnited(world, blockpos1, blockpos, org.loomdev.api.event.block.BlockIgnitedEvent.Cause.LAVA).isCancelled()) {
+                                return;
+                            }
+                            // Loom end
                             world.setBlockState(blockpos1, AbstractFireBlock.getState(world, blockpos1));
                             return;
                         }
@@ -66,6 +71,11 @@
                     }
 
                     if (world.isAir(blockpos2.up()) && this.hasBurnableBlock(world, blockpos2)) {
+                        // Loom start :: fire BlockIgnitedEvent
+                        if (org.loomdev.loom.event.LoomEventDispatcher.onBlockIgnited(world, blockpos2.up(), blockpos, org.loomdev.api.event.block.BlockIgnitedEvent.Cause.LAVA).isCancelled()) {
+                            return;
+                        }
+                        // Loom end
                         world.setBlockState(blockpos2.up(), AbstractFireBlock.getState(world, blockpos2));
                     }
                 }
