--- a/net/minecraft/world/entity/projectile/AbstractArrow.java
+++ b/net/minecraft/world/entity/projectile/AbstractArrow.java
@@ -54,18 +54,19 @@
     private static final int FLAG_CROSSBOW = 4;
     @Nullable
     private BlockState lastState;
-    protected boolean inGround;
+    public boolean inGround; // Loom :: protected -> public (AbstractArrowImpl)
     protected int inGroundTime;
     public AbstractArrow.Pickup pickup;
     public int shakeTime;
     private int life;
     private double baseDamage;
-    private int knockback;
+    public int knockback; // Loom :: private -> public (AbstractArrowImpl)
     private SoundEvent soundEvent;
     @Nullable
     private IntOpenHashSet piercingIgnoreEntityIds;
     @Nullable
     private List<Entity> piercedAndKilledEntities;
+    public float waterInertia = 0.6F; // Loom :: (AbstractArrowImpl)
 
     protected AbstractArrow(EntityType<? extends AbstractArrow> entitytype, Level level) {
         super(entitytype, level);
@@ -524,7 +525,7 @@
         }
     }
 
-    protected abstract ItemStack getPickupItem();
+    public abstract ItemStack getPickupItem(); // Loom :: protected -> public (AbstractArrowImpl)
 
     protected Entity.MovementEmission getMovementEmission() {
         return Entity.MovementEmission.NONE;
@@ -609,7 +610,7 @@
     }
 
     protected float getWaterInertia() {
-        return 0.6F;
+        return waterInertia; // Loom :: (AbstractArrowImpl)
     }
 
     public void setNoPhysics(boolean flag) {
