--- a/net/minecraft/world/entity/projectile/LargeFireball.java
+++ b/net/minecraft/world/entity/projectile/LargeFireball.java
@@ -13,7 +13,7 @@
 
 public class LargeFireball extends Fireball {
 
-    public int explosionPower = 1;
+    public int explosionPower = 1; // Loom :: private -> public (LargeFireballImpl)
 
     public LargeFireball(EntityType<? extends LargeFireball> entitytype, Level level) {
         super(entitytype, level);
@@ -26,7 +26,7 @@
     protected void onHit(HitResult hitresult) {
         super.onHit(hitresult);
         if (!this.level.isClientSide) {
-            boolean flag = this.level.getGameRules().getBoolean(GameRules.RULE_MOBGRIEFING);
+            boolean flag = this.level.getGameRules().getBoolean(GameRules.RULE_MOBGRIEFING) && explosionPower != -1F; // Loom :: (LargeFireballImpl)
 
             this.level.explode((Entity) null, this.getX(), this.getY(), this.getZ(), (float) this.explosionPower, flag, flag ? Explosion.BlockInteraction.DESTROY : Explosion.BlockInteraction.NONE);
             this.discard();
