--- a/net/minecraft/world/entity/projectile/ShulkerBullet.java
+++ b/net/minecraft/world/entity/projectile/ShulkerBullet.java
@@ -32,7 +32,7 @@
 
     private static final double SPEED = 0.15D;
     @Nullable
-    private Entity finalTarget;
+    public Entity finalTarget; // Loom :: private -> public (ShulkerBulletImpl)
     @Nullable
     private Direction currentMoveDirection;
     private int flightSteps;
@@ -40,7 +40,7 @@
     private double targetDeltaY;
     private double targetDeltaZ;
     @Nullable
-    private UUID targetId;
+    public UUID targetId; // Loom :: private -> public (ShulkerBulletImpl)
 
     public ShulkerBullet(EntityType<? extends ShulkerBullet> entitytype, Level level) {
         super(entitytype, level);
