--- a/net/minecraft/entity/ExperienceOrbEntity.java
+++ b/net/minecraft/entity/ExperienceOrbEntity.java
@@ -20,9 +20,9 @@
     public int renderTicks;
     public int orbAge;
     public int pickupDelay;
-    private int health;
-    private int amount;
-    private PlayerEntity target;
+    public int health; // Loom :: public accessor
+    public int amount; // Loom :: public accessor
+    public PlayerEntity target; // Loom :: public accessor
     private int lastTargetUpdateTick;
 
     public ExperienceOrbEntity(World world, double d0, double d1, double d2, int i) {
