--- a/net/minecraft/entity/ItemEntity.java
+++ b/net/minecraft/entity/ItemEntity.java
@@ -32,7 +32,7 @@
 
     private static final TrackedData<ItemStack> STACK = DataTracker.registerData(ItemEntity.class, TrackedDataHandlerRegistry.ITEM_STACK);
     private int age;
-    private int pickupDelay;
+    public int pickupDelay; // Loom :: public accessor
     private int health;
     private UUID thrower;
     private UUID owner;
