--- a/net/minecraft/entity/LightningEntity.java
+++ b/net/minecraft/entity/LightningEntity.java
@@ -25,7 +25,7 @@
     private int ambientTick;
     public long seed;
     private int remainingActions;
-    private boolean cosmetic;
+    public boolean cosmetic; // Loom :: public access
     @Nullable
     private ServerPlayerEntity channeler;
 
@@ -102,7 +102,13 @@
             BlockState blockstate = AbstractFireBlock.getState(this.world, blockpos);
 
             if (this.world.getBlockState(blockpos).isAir() && blockstate.canPlaceAt(this.world, blockpos)) {
+                // Loom start :: fire BlockIgnitedEvent
+                if (org.loomdev.loom.event.LoomEventDispatcher.onBlockIgnited(this.world, blockpos, this, org.loomdev.api.event.block.BlockIgnitedEvent.Cause.LIGHTNING).isCancelled()) {
+                    return;
+                }
+
                 this.world.setBlockState(blockpos, blockstate);
+                // Loom end
             }
 
             for (int j = 0; j < i; ++j) {
@@ -110,7 +116,13 @@
 
                 blockstate = AbstractFireBlock.getState(this.world, blockpos1);
                 if (this.world.getBlockState(blockpos1).isAir() && blockstate.canPlaceAt(this.world, blockpos1)) {
+                    // Loom start :: fire BlockIgnitedEvent
+                    if (org.loomdev.loom.event.LoomEventDispatcher.onBlockIgnited(this.world, blockpos1, this, org.loomdev.api.event.block.BlockIgnitedEvent.Cause.LIGHTNING).isCancelled()) {
+                        return;
+                    }
+
                     this.world.setBlockState(blockpos1, blockstate);
+                    // Loom end
                 }
             }
 
