--- a/net/minecraft/entity/TntEntity.java
+++ b/net/minecraft/entity/TntEntity.java
@@ -17,6 +17,7 @@
     @Nullable
     private LivingEntity causingEntity;
     private int fuseTimer;
+    public float explosionPower = 4.0F; // Loom :: overridable explosion radius
 
     public TntEntity(EntityType<? extends TntEntity> entitytype, World world) {
         super(entitytype, world);
@@ -78,7 +79,7 @@
     private void explode() {
         float f = 4.0F;
 
-        this.world.createExplosion(this, this.getX(), this.getBodyY(0.0625D), this.getZ(), 4.0F, Explosion.DestructionType.BREAK);
+        this.world.createExplosion(this, this.getX(), this.getBodyY(0.0625D), this.getZ(), this.explosionPower, Explosion.DestructionType.BREAK); // Loom :: overridable explosion radius
     }
 
     protected void writeCustomDataToTag(CompoundTag compoundtag) {
