--- a/net/minecraft//world/entity/animal/goat/Goat.java
+++ b/net/minecraft//world/entity/animal/goat/Goat.java
@@ -116,7 +116,7 @@
     }
 
     public Brain<Goat> getBrain() {
-        return super.getBrain();
+        return (Brain<Goat>) super.getBrain(); // Loom :: decompile error
     }
 
     protected void customServerAiStep() {
@@ -225,9 +225,11 @@
         return new Goat.GoatPathNavigation(this, level);
     }
 
+    /* // Loom start :: decompile error
     public AgeableMob getBreedOffspring(ServerLevel serverlevel, AgeableMob ageablemob) {
         return this.getBreedOffspring(serverlevel, ageablemob);
     }
+     */ // Loom end
 
     static class GoatNodeEvaluator extends WalkNodeEvaluator {
 
