--- a/net/minecraft/server/world/ServerWorld.java
+++ b/net/minecraft/server/world/ServerWorld.java
@@ -149,7 +149,7 @@
     private final ServerChunkManager serverChunkManager;
     boolean inEntityTick;
     private final MinecraftServer server;
-    private final ServerWorldProperties field_24456;
+    public final ServerWorldProperties field_24456; // Loom :: public access
     public boolean savingDisabled;
     private boolean allPlayersSleeping;
     private int idleTimeout;
@@ -443,7 +443,7 @@
 
     private void wakeSleepingPlayers() {
         ((List) this.players.stream().filter(LivingEntity::isSleeping).collect(Collectors.toList())).forEach((serverplayerentity) -> {
-            serverplayerentity.wakeUp(false, false);
+            ((ServerPlayerEntity) serverplayerentity).wakeUp(false, false); // Loom :: decompile error
         });
     }
 
@@ -1528,19 +1528,5 @@
         });
     }
 
-    public Scoreboard getScoreboard() {
-        return this.getScoreboard();
-    }
-
-    public ChunkManager getChunkManager() {
-        return this.getChunkManager();
-    }
-
-    public TickScheduler getFluidTickScheduler() {
-        return this.getFluidTickScheduler();
-    }
-
-    public TickScheduler getBlockTickScheduler() {
-        return this.getBlockTickScheduler();
-    }
+    // Loom :: decompile error
 }
