--- a/net/minecraft/world/entity/decoration/ItemFrame.java
+++ b/net/minecraft/world/entity/decoration/ItemFrame.java
@@ -40,8 +40,8 @@
     private static final Logger LOGGER = LogManager.getLogger();
     private static final EntityDataAccessor<ItemStack> DATA_ITEM = SynchedEntityData.defineId(ItemFrame.class, EntityDataSerializers.ITEM_STACK);
     private static final EntityDataAccessor<Integer> DATA_ROTATION = SynchedEntityData.defineId(ItemFrame.class, EntityDataSerializers.INT);
-    private float dropChance;
-    private boolean fixed;
+    public float dropChance; // Loom :: private -> public (ItemFrameImpl)
+    public boolean fixed; // Loom :: private -> public (ItemFrameImpl)
 
     public ItemFrame(EntityType<? extends ItemFrame> entitytype, Level level) {
         super(entitytype, level);
@@ -67,7 +67,7 @@
         this.getEntityData().define(ItemFrame.DATA_ROTATION, 0);
     }
 
-    protected void setDirection(Direction direction) {
+    public void setDirection(Direction direction) { // Loom :: protected -> public (ItemFrameImpl)
         Validate.notNull(direction);
         this.direction = direction;
         if (direction.getAxis().isHorizontal()) {
