--- a/net/minecraft//world/item/BlockItem.java
+++ b/net/minecraft//world/item/BlockItem.java
@@ -65,6 +65,13 @@
                 return InteractionResult.FAIL;
             } else {
                 BlockState blockstate = this.getPlacementState(blockplacecontext1);
+                // Loom start :: fire BlockEvent.Place
+                if (blockstate != null && blockplacecontext1.getPlayer() != null) {
+                    var event = org.loomdev.loom.event.LoomEventDispatcher.fire(new org.loomdev.loom.event.block.BlockEventImpl.PlaceImpl(blockplacecontext1.getLevel(), blockplacecontext1.getClickedPos(), blockplacecontext1.getPlayer(), blockstate));
+                    if (event.isCanceled()) return InteractionResult.FAIL;
+                    blockstate = event.getMinecraftState();
+                }
+                // Loom end
 
                 if (blockstate == null) {
                     return InteractionResult.FAIL;
@@ -238,7 +245,7 @@
 
             if (compoundtag != null) {
                 ListTag listtag = compoundtag.getCompound("BlockEntityTag").getList("Items", 10);
-                Stream stream = listtag.stream();
+                Stream<net.minecraft.nbt.Tag> stream = listtag.stream(); // Loom :: decompile error
 
                 CompoundTag.class.getClass();
                 ItemUtils.onContainerDestroyed(itementity, stream.map(CompoundTag.class::cast).map(ItemStack::of));
