--- a/net/minecraft/world/item/FireChargeItem.java
+++ b/net/minecraft/world/item/FireChargeItem.java
@@ -30,11 +30,13 @@
         if (!CampfireBlock.canLight(blockstate) && !CandleBlock.canLight(blockstate) && !CandleCakeBlock.canLight(blockstate)) {
             blockpos = blockpos.relative(useoncontext.getClickedFace());
             if (BaseFireBlock.canBePlacedAt(level, blockpos, useoncontext.getHorizontalDirection())) {
+                // TODO if (useoncontext.getPlayer() != null && org.loomdev.loom.event.LoomEventDispatcher.onBlockIgnite(level, blockpos, useoncontext.getPlayer(), org.loomdev.api.event.block.BlockIgniteEvent.Cause.FIRE_CHARGE).isCancelled()) return InteractionResult.FAIL; // Loom :: fire BlockIgniteEvent
                 this.playSound(level, blockpos);
                 level.setBlockAndUpdate(blockpos, BaseFireBlock.getState(level, blockpos));
                 flag = true;
             }
         } else {
+            // TODO if (useoncontext.getPlayer() != null && org.loomdev.loom.event.LoomEventDispatcher.onBlockIgnite(level, blockpos, useoncontext.getPlayer(), org.loomdev.api.event.block.BlockIgniteEvent.Cause.FIRE_CHARGE).isCancelled()) return InteractionResult.FAIL; // Loom :: fire BlockIgniteEvent
             this.playSound(level, blockpos);
             level.setBlockAndUpdate(blockpos, (BlockState) blockstate.setValue(BlockStateProperties.LIT, true));
             flag = true;
