--- a/net/minecraft/world/item/FlintAndSteelItem.java
+++ b/net/minecraft/world/item/FlintAndSteelItem.java
@@ -33,6 +33,7 @@
             BlockPos blockpos1 = blockpos.relative(useoncontext.getClickedFace());
 
             if (BaseFireBlock.canBePlacedAt(level, blockpos1, useoncontext.getHorizontalDirection())) {
+                // TODO if (player != null && org.loomdev.loom.event.LoomEventDispatcher.onBlockIgnite(level, blockpos, player, org.loomdev.api.event.block.BlockIgniteEvent.Cause.FLINT_AND_STEEL).isCancelled()) return InteractionResult.FAIL; // Loom :: fire BlockIgniteEvent
                 level.playSound(player, blockpos1, SoundEvents.FLINTANDSTEEL_USE, SoundSource.BLOCKS, 1.0F, level.getRandom().nextFloat() * 0.4F + 0.8F);
                 BlockState blockstate1 = BaseFireBlock.getState(level, blockpos1);
 
@@ -52,6 +53,7 @@
                 return InteractionResult.FAIL;
             }
         } else {
+            // TODO if (player != null && org.loomdev.loom.event.LoomEventDispatcher.onBlockIgnite(level, blockpos, player, org.loomdev.api.event.block.BlockIgniteEvent.Cause.FLINT_AND_STEEL).isCancelled()) return InteractionResult.FAIL; // Loom :: fire BlockIgniteEvent
             level.playSound(player, blockpos, SoundEvents.FLINTANDSTEEL_USE, SoundSource.BLOCKS, 1.0F, level.getRandom().nextFloat() * 0.4F + 0.8F);
             level.setBlock(blockpos, (BlockState) blockstate.setValue(BlockStateProperties.LIT, true), 11);
             level.gameEvent(player, GameEvent.BLOCK_PLACE, blockpos);
