--- a/net/minecraft//world/item/ItemStack.java
+++ b/net/minecraft//world/item/ItemStack.java
@@ -107,7 +107,7 @@
     private int count;
     private int popTime;
     @Deprecated
-    private final Item item;
+    public Item item; // Loom :: private -> public, non-final
     private CompoundTag tag;
     private boolean emptyCacheFlag;
     private Entity entityRepresentation;
@@ -139,7 +139,7 @@
         this.updateEmptyCacheFlag();
     }
 
-    private void updateEmptyCacheFlag() {
+    public void updateEmptyCacheFlag() { // Loom :: private -> public
         this.emptyCacheFlag = false;
         this.emptyCacheFlag = this.isEmpty();
     }
