--- a/net/minecraft/world/level/block/entity/SculkSensorBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/SculkSensorBlockEntity.java
@@ -52,6 +52,7 @@
         BlockState blockstate = this.getBlockState();
 
         if (!level.isClientSide() && SculkSensorBlock.canActivate(blockstate)) {
+            if (org.loomdev.loom.event.LoomEventDispatcher.onSculkSensorActivate(level, getBlockPos(), gameeventlistener.getListenerSource(), gameevent).isCanceled()) return; // Loom :: fire SculkSensorEvent.Activate
             this.lastVibrationFrequency = SculkSensorBlock.VIBRATION_STRENGTH_FOR_EVENT.getInt(gameevent);
             SculkSensorBlock.activate(level, this.worldPosition, blockstate, getRedstoneStrengthForDistance(i, gameeventlistener.getListenerRadius()));
         }
