--- a/net/minecraft/network/protocol/game/ClientboundTabListPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundTabListPacket.java
@@ -13,6 +13,13 @@
 
     public ClientboundTabListPacket() {}
 
+    // Loom start :: (PlayerImpl)
+    public ClientboundTabListPacket(Component header, Component footer) {
+        this.header = header;
+        this.footer = footer;
+    }
+    // Loom end
+
     public void read(FriendlyByteBuf friendlybytebuf) throws IOException {
         this.header = friendlybytebuf.readComponent();
         this.footer = friendlybytebuf.readComponent();
@@ -27,7 +34,9 @@
         clientgamepacketlistener.handleTabListCustomisation(this);
     }
 
+    /* // Loom start :: decompile error
     public void handle(PacketListener packetlistener) {
         this.handle((ClientGamePacketListener) packetlistener);
     }
+     */ // Loom end
 }
