--- a/net/minecraft//network/protocol/game/ClientboundGameEventPacket.java
+++ b/net/minecraft//network/protocol/game/ClientboundGameEventPacket.java
@@ -25,7 +25,7 @@
     public static final int DEMO_PARAM_HINT_2 = 102;
     public static final int DEMO_PARAM_HINT_3 = 103;
     public static final int DEMO_PARAM_HINT_4 = 104;
-    private final ClientboundGameEventPacket.Type event;
+    public final ClientboundGameEventPacket.Type event; // Loom :: private -> public (PlayerPacketFilter)
     private final float param;
 
     public ClientboundGameEventPacket(ClientboundGameEventPacket.Type clientboundgameeventpacket_type, float f) {
@@ -55,9 +55,7 @@
         return this.param;
     }
 
-    public void handle(PacketListener packetlistener) {
-        this.handle((ClientGamePacketListener) packetlistener);
-    }
+    // Loom :: decompile error
 
     public static class Type {
 
