--- a/net/minecraft/network/protocol/game/ClientboundTabListPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundTabListPacket.java
@@ -15,6 +15,13 @@
         this.footer = friendlybytebuf.readComponent();
     }
 
+    // Loom start :: (PlayerImpl)
+    public ClientboundTabListPacket(Component header, Component footer) {
+        this.header = header;
+        this.footer = footer;
+    }
+    // Loom end
+
     public void write(FriendlyByteBuf friendlybytebuf) {
         friendlybytebuf.writeComponent(this.header);
         friendlybytebuf.writeComponent(this.footer);
@@ -24,7 +31,9 @@
         clientgamepacketlistener.handleTabListCustomisation(this);
     }
 
+    /* // Loom start :: decompile error
     public void handle(PacketListener packetlistener) {
         this.handle((ClientGamePacketListener) packetlistener);
     }
+     */ // Loom end
 }
