--- a/net/minecraft//network/protocol/game/ServerboundClientInformationPacket.java
+++ b/net/minecraft//network/protocol/game/ServerboundClientInformationPacket.java
@@ -10,7 +10,7 @@
 
     public static final int MAX_LANGUAGE_LENGTH = 16;
     private final String language;
-    private final int viewDistance;
+    public final int viewDistance; // Loom :: private -> public
     private final ChatVisiblity chatVisibility;
     private final boolean chatColors;
     private final int modelCustomisation;
@@ -79,7 +79,9 @@
         return this.textFilteringEnabled;
     }
 
+    /* // Loom start :: decompile error
     public void handle(PacketListener packetlistener) {
         this.handle((ServerGamePacketListener) packetlistener);
     }
+     */ // Loom end
 }
