--- a/net/minecraft//world/entity/npc/WanderingTrader.java
+++ b/net/minecraft//world/entity/npc/WanderingTrader.java
@@ -49,8 +49,8 @@
 
     private static final int NUMBER_OF_TRADE_OFFERS = 5;
     @Nullable
-    private BlockPos wanderTarget;
-    private int despawnDelay;
+    public BlockPos wanderTarget; // Loom :: private -> public
+    public int despawnDelay; // Loom :: private -> public
 
     public WanderingTrader(EntityType<? extends WanderingTrader> entitytype, Level level) {
         super(entitytype, level);
