--- a/net/minecraft//world/level/material/FlowingFluid.java
+++ b/net/minecraft//world/level/material/FlowingFluid.java
@@ -434,13 +434,23 @@
             FluidState fluidstate1 = this.getNewLiquid(level, blockpos, level.getBlockState(blockpos));
             int i = this.getSpreadDelay(level, blockpos, fluidstate, fluidstate1);
 
+
             if (fluidstate1.isEmpty()) {
                 fluidstate = fluidstate1;
-                level.setBlock(blockpos, Blocks.AIR.defaultBlockState(), 3);
+                // Loom start :: fire FluidEvent.LevelChange
+                var event = org.loomdev.loom.event.LoomEventDispatcher.fire(new org.loomdev.loom.event.fluid.FluidEventImpl.LevelChange(level, blockpos, Blocks.AIR.defaultBlockState()));
+                if (event.isCanceled()) return;
+                level.setBlock(blockpos, event.getMinecraftState(), 3);
+                // Loom end
             } else if (!fluidstate1.equals(fluidstate)) {
                 fluidstate = fluidstate1;
                 BlockState blockstate = fluidstate1.createLegacyBlock();
 
+                // Loom start :: fire FluidEvent.LevelChange
+                var event = org.loomdev.loom.event.LoomEventDispatcher.fire(new org.loomdev.loom.event.fluid.FluidEventImpl.LevelChange(level, blockpos, blockstate));
+                if (event.isCanceled()) return;
+                blockstate = ((org.loomdev.loom.block.BlockStateImpl) event.getNewState()).getMinecraftState();
+                // Loom end
                 level.setBlock(blockpos, blockstate, 2);
                 level.getLiquidTicks().scheduleTick(blockpos, fluidstate1.getType(), i);
                 level.updateNeighborsAt(blockpos, blockstate.getBlock());
