--- a/net/minecraft/entity/projectile/thrown/ThrownItemEntity.java
+++ b/net/minecraft/entity/projectile/thrown/ThrownItemEntity.java
@@ -13,7 +13,7 @@
 
 public abstract class ThrownItemEntity extends ThrownEntity {
 
-    private static final TrackedData<ItemStack> ITEM = DataTracker.registerData(ThrownItemEntity.class, TrackedDataHandlerRegistry.ITEM_STACK);
+    public static final TrackedData<ItemStack> ITEM = DataTracker.registerData(ThrownItemEntity.class, TrackedDataHandlerRegistry.ITEM_STACK); // Loom :: public accessor
 
     public ThrownItemEntity(EntityType<? extends ThrownItemEntity> entitytype, World world) {
         super(entitytype, world);
