--- a/net/minecraft/world/inventory/AbstractFurnaceMenu.java
+++ b/net/minecraft/world/inventory/AbstractFurnaceMenu.java
@@ -142,7 +142,7 @@
         return this.level.getRecipeManager().getRecipeFor(this.recipeType, new SimpleContainer(new ItemStack[]{itemstack}), this.level).isPresent();
     }
 
-    protected boolean isFuel(ItemStack itemstack) {
+    public boolean isFuel(ItemStack itemstack) { // Loom :: protected -> public
         return AbstractFurnaceBlockEntity.isFuel(itemstack);
     }
 
