--- a/net/minecraft/entity/raid/RaiderEntity.java
+++ b/net/minecraft/entity/raid/RaiderEntity.java
@@ -48,7 +48,7 @@
 
 public abstract class RaiderEntity extends PatrolEntity {
 
-    protected static final TrackedData<Boolean> CELEBRATING = DataTracker.registerData(RaiderEntity.class, TrackedDataHandlerRegistry.BOOLEAN);
+    public static final TrackedData<Boolean> CELEBRATING = DataTracker.registerData(RaiderEntity.class, TrackedDataHandlerRegistry.BOOLEAN); // Loom :: public accessor
     private static final Predicate<ItemEntity> OBTAINABLE_OMINOUS_BANNER_PREDICATE = (itementity) -> {
         return !itementity.cannotPickup() && itementity.isAlive() && ItemStack.areEqual(itementity.getStack(), Raid.getOminousBanner());
     };
