--- a/net/minecraft/village/TradeOffer.java
+++ b/net/minecraft/village/TradeOffer.java
@@ -7,16 +7,16 @@
 
 public class TradeOffer {
 
-    private final ItemStack firstBuyItem;
-    private final ItemStack secondBuyItem;
-    private final ItemStack sellItem;
-    private int uses;
-    private final int maxUses;
-    private boolean rewardingPlayerExperience;
-    private int specialPrice;
-    private int demandBonus;
-    private float priceMultiplier;
-    private int traderExperience;
+    public ItemStack firstBuyItem; // Loom :: public accessor, non-final
+    public ItemStack secondBuyItem; // Loom :: public accessor, non-final
+    public ItemStack sellItem; // Loom :: public accessor, non-final
+    public int uses; // Loom :: public accessor
+    public int maxUses; // Loom :: public accessor, non-final
+    public boolean rewardingPlayerExperience; // Loom :: public accessor
+    public int specialPrice; // Loom :: public accessor
+    public int demandBonus; // Loom :: public accessor
+    public float priceMultiplier; // Loom :: public accessor
+    public int traderExperience; // Loom :: public accessor
 
     public TradeOffer(CompoundTag compoundtag) {
         this.rewardingPlayerExperience = true;
