--- a/net/minecraft/server/players/PlayerList.java
+++ b/net/minecraft/server/players/PlayerList.java
@@ -196,7 +196,12 @@
             translatablecomponent = new TranslatableComponent("multiplayer.player.joined.renamed", new Object[]{serverplayer.getDisplayName(), s});
         }
 
-        this.broadcastMessage(translatablecomponent.withStyle(ChatFormatting.YELLOW), ChatType.SYSTEM, Util.NIL_UUID);
+        // Loom start :: fire PlayerJoinEvent
+        var event = org.loomdev.loom.event.LoomEventDispatcher.fire(new org.loomdev.loom.event.entity.PlayerEventImpl.JoinImpl(serverplayer, translatablecomponent.withStyle(ChatFormatting.YELLOW)));
+        if (event.getJoinMessage().toString().length() != 0) {
+            this.broadcastMessage(org.loomdev.loom.util.transformer.TextTransformer.toMinecraft(event.getJoinMessage()), ChatType.SYSTEM, Util.NIL_UUID);
+        }
+        // Loom end
         servergamepacketlistenerimpl.teleport(serverplayer.getX(), serverplayer.getY(), serverplayer.getZ(), serverplayer.yRot, serverplayer.xRot);
         this.players.add(serverplayer);
         this.playersByUUID.put(serverplayer.getUUID(), serverplayer);
@@ -549,6 +554,7 @@
         while (iterator.hasNext()) {
             ServerPlayer serverplayer = (ServerPlayer) iterator.next();
 
+            if (org.loomdev.loom.entity.player.PlayerPacketFilter.shouldSend(serverplayer, packet)) // Loom :: player packet filter
             serverplayer.connection.send(packet);
         }
 
@@ -561,6 +567,7 @@
             ServerPlayer serverplayer = (ServerPlayer) iterator.next();
 
             if (serverplayer.level.dimension() == resourcekey) {
+                if (org.loomdev.loom.entity.player.PlayerPacketFilter.shouldSend(serverplayer, packet)) // Loom :: player packet filter
                 serverplayer.connection.send(packet);
             }
         }
@@ -632,7 +639,7 @@
     }
 
     public void deop(GameProfile gameprofile) {
-        this.ops.remove((Object) gameprofile);
+        this.ops.remove(gameprofile); // Loom :: decompile error
         ServerPlayer serverplayer = this.getPlayer(gameprofile.getId());
 
         if (serverplayer != null) {
