--- a/net/minecraft//server/players/PlayerList.java
+++ b/net/minecraft//server/players/PlayerList.java
@@ -93,6 +93,7 @@
 import net.minecraft.world.scores.Team;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.loomdev.loom.transformer.Transformer;
 
 public abstract class PlayerList {
 
@@ -201,7 +202,12 @@
             translatablecomponent = new TranslatableComponent("multiplayer.player.joined.renamed", new Object[]{serverplayer.getDisplayName(), s});
         }
 
-        this.broadcastMessage(translatablecomponent.withStyle(ChatFormatting.YELLOW), ChatType.SYSTEM, Util.NIL_UUID);
+        // Loom start :: fire PlayerJoinEvent
+        var event = org.loomdev.loom.event.LoomEventDispatcher.fire(new org.loomdev.loom.event.entity.PlayerEventImpl.JoinImpl(serverplayer, translatablecomponent.withStyle(ChatFormatting.YELLOW)));
+        if (event.getJoinMessage().toString().length() != 0) {
+            this.broadcastMessage(Transformer.COMPONENT.toMinecraft(event.getJoinMessage()), ChatType.SYSTEM, Util.NIL_UUID);
+        }
+        // Loom end
         servergamepacketlistenerimpl.teleport(serverplayer.getX(), serverplayer.getY(), serverplayer.getZ(), serverplayer.getYRot(), serverplayer.getXRot());
         this.players.add(serverplayer);
         this.playersByUUID.put(serverplayer.getUUID(), serverplayer);
@@ -554,6 +560,7 @@
         while (iterator.hasNext()) {
             ServerPlayer serverplayer = (ServerPlayer) iterator.next();
 
+            if (org.loomdev.loom.entity.player.PlayerPacketFilter.shouldSend(serverplayer, packet)) // Loom :: player packet filter
             serverplayer.connection.send(packet);
         }
 
@@ -566,6 +573,7 @@
             ServerPlayer serverplayer = (ServerPlayer) iterator.next();
 
             if (serverplayer.level.dimension() == resourcekey) {
+                if (org.loomdev.loom.entity.player.PlayerPacketFilter.shouldSend(serverplayer, packet)) // Loom :: player packet filter
                 serverplayer.connection.send(packet);
             }
         }
@@ -637,7 +645,7 @@
     }
 
     public void deop(GameProfile gameprofile) {
-        this.ops.remove((Object) gameprofile);
+        this.ops.remove(gameprofile); // Loom :: decompile error
         ServerPlayer serverplayer = this.getPlayer(gameprofile.getId());
 
         if (serverplayer != null) {
