--- a/net/minecraft/block/BedBlock.java
+++ b/net/minecraft/block/BedBlock.java
@@ -135,7 +135,18 @@
         if (vec3d.y < 0.0D) {
             double d0 = entity instanceof LivingEntity ? 1.0D : 0.8D;
 
-            entity.setVelocity(vec3d.x, -vec3d.y * 0.6600000262260437D * d0, vec3d.z);
+            // Loom :: fire BlockBouncedEvent
+            org.loomdev.loom.event.LoomEventDispatcher.onBlockBounced(entity.world, entity.getBlockPos(), entity, d0).thenAccept(event -> {
+               if (!event.isCancelled()) {
+                   entity.setVelocity(vec3d.x, -vec3d.y * 0.6600000262260437D * event.getBounceMultiplier(), vec3d.z);
+               } else {
+                   // If cancelled, make sure the player doesn't bounce
+                   entity.setVelocity(0, 0, 0);
+               }
+
+               entity.velocityModified = true;
+            });
+            // Loom end;
         }
 
     }
