--- a/net/minecraft/block/CampfireBlock.java
+++ b/net/minecraft/block/CampfireBlock.java
@@ -173,8 +173,13 @@
 
             if (flag && !(Boolean) blockstate.get(CampfireBlock.LIT) && !(Boolean) blockstate.get(CampfireBlock.WATERLOGGED)) {
                 BlockPos blockpos = blockhitresult.getBlockPos();
+                // Loom start :: fire BlockIgnitedEvent
+                if (org.loomdev.loom.event.LoomEventDispatcher.onBlockIgnited(world, blockpos, projectileentity, org.loomdev.api.event.block.BlockIgnitedEvent.Cause.FLAME_ARROW).isCancelled()) {{
+                    return;
+                }}
 
                 world.setBlockState(blockpos, (BlockState) blockstate.with(Properties.LIT, true), 11);
+                // Loom end
             }
         }
 
