--- a/net/minecraft/block/CoralBlockBlock.java
+++ b/net/minecraft/block/CoralBlockBlock.java
@@ -22,7 +22,13 @@
 
     public void scheduledTick(BlockState blockstate, ServerWorld serverworld, BlockPos blockpos, Random random) {
         if (!this.isInWater(serverworld, blockpos)) {
+            // Loom start :: fire CoralDiedEvent
+            org.loomdev.loom.event.LoomEventDispatcher.onPlantDied(serverworld, blockpos).thenAccept(event -> {
+                if (!event.isCancelled()) {
             serverworld.setBlockState(blockpos, this.deadCoralBlock.getDefaultState(), 2);
+                }
+            });
+            // Loom end
         }
 
     }
