--- a/net/minecraft/block/CoralWallFanBlock.java
+++ b/net/minecraft/block/CoralWallFanBlock.java
@@ -23,7 +23,13 @@
 
     public void scheduledTick(BlockState blockstate, ServerWorld serverworld, BlockPos blockpos, Random random) {
         if (!isInWater(blockstate, serverworld, blockpos)) {
+            // Loom start :: fire CoralDiedEvent
+            org.loomdev.loom.event.LoomEventDispatcher.onPlantDied(serverworld, blockpos).thenAccept(event -> {
+               if (!event.isCancelled()) {
             serverworld.setBlockState(blockpos, (BlockState) ((BlockState) this.deadCoralBlock.getDefaultState().with(CoralWallFanBlock.WATERLOGGED, false)).with(CoralWallFanBlock.FACING, blockstate.get(CoralWallFanBlock.FACING)), 2);
+               }
+            });
+            // Loom end
         }
 
     }
