--- a/net/minecraft/block/CoralBlock.java
+++ b/net/minecraft/block/CoralBlock.java
@@ -26,7 +26,13 @@
 
     public void scheduledTick(BlockState blockstate, ServerWorld serverworld, BlockPos blockpos, Random random) {
         if (!isInWater(blockstate, serverworld, blockpos)) {
+            // Loom start :: fire CoralDiedEvent
+            org.loomdev.loom.event.LoomEventDispatcher.onCoralDied(serverworld, blockpos).thenAccept(event -> {
+                if (!event.isCancelled()) {
             serverworld.setBlockState(blockpos, (BlockState) this.deadCoralBlock.getDefaultState().with(CoralBlock.WATERLOGGED, false), 2);
+                }
+            });
+            // Loom end
         }
 
     }
