--- a/net/minecraft/block/LeavesBlock.java
+++ b/net/minecraft/block/LeavesBlock.java
@@ -35,8 +35,14 @@
 
     public void randomTick(BlockState blockstate, ServerWorld serverworld, BlockPos blockpos, Random random) {
         if (!(Boolean) blockstate.get(LeavesBlock.PERSISTENT) && (Integer) blockstate.get(LeavesBlock.DISTANCE) == 7) {
+            // Loom start :: fire LeavesDecayedEvent
+            org.loomdev.loom.event.LoomEventDispatcher.onPlantDecayed(serverworld, blockpos).thenAccept(event -> {
+               if (!event.isCancelled()) {
             dropStacks(blockstate, serverworld, blockpos);
             serverworld.removeBlock(blockpos, false);
+               }
+            });
+            // Loom end
         }
 
     }
