--- a/net/minecraft//network/protocol/status/ServerStatus.java
+++ b/net/minecraft//network/protocol/status/ServerStatus.java
@@ -14,7 +14,7 @@
 import net.minecraft.network.chat.Component;
 import net.minecraft.util.GsonHelper;
 
-public class ServerStatus {
+public class ServerStatus implements Cloneable { // Loom
 
     public static final int FAVICON_WIDTH = 64;
     public static final int FAVICON_HEIGHT = 64;
@@ -59,8 +59,8 @@
 
     public static class Players {
 
-        private final int maxPlayers;
-        private final int numPlayers;
+        public int maxPlayers; // Loom :: private, final -> public (StatusPingEventImpl)
+        public int numPlayers; // Loom :: private, final -> public (StatusPingEventImpl)
         private GameProfile[] sample;
 
         public Players(int i, int j) {
@@ -135,6 +135,7 @@
                 return jsonobject;
             }
 
+            /* // Loom start :: decompile error
             public JsonElement serialize(Object object, Type type, JsonSerializationContext jsonserializationcontext) {
                 return this.serialize((ServerStatus.Players) object, type, jsonserializationcontext);
             }
@@ -142,13 +143,14 @@
             public Object deserialize(JsonElement jsonelement, Type type, JsonDeserializationContext jsondeserializationcontext) throws JsonParseException {
                 return this.deserialize(jsonelement, type, jsondeserializationcontext);
             }
+             */ // Loom end
         }
     }
 
     public static class Version {
 
-        private final String name;
-        private final int protocol;
+        public String name; // Loom :: private, final -> public (StatusPingEventImpl)
+        public int protocol; // Loom :: private, final -> public (StatusPingEventImpl)
 
         public Version(String s, int i) {
             this.name = s;
@@ -181,6 +183,7 @@
                 return jsonobject;
             }
 
+            /* // Loom start :: decompile error
             public JsonElement serialize(Object object, Type type, JsonSerializationContext jsonserializationcontext) {
                 return this.serialize((ServerStatus.Version) object, type, jsonserializationcontext);
             }
@@ -188,6 +191,7 @@
             public Object deserialize(JsonElement jsonelement, Type type, JsonDeserializationContext jsondeserializationcontext) throws JsonParseException {
                 return this.deserialize(jsonelement, type, jsondeserializationcontext);
             }
+             */ // Loom end
         }
     }
 
@@ -240,6 +244,7 @@
             return jsonobject;
         }
 
+        /* // Loom start :: decompile error
         public JsonElement serialize(Object object, Type type, JsonSerializationContext jsonserializationcontext) {
             return this.serialize((ServerStatus) object, type, jsonserializationcontext);
         }
@@ -247,5 +252,13 @@
         public Object deserialize(JsonElement jsonelement, Type type, JsonDeserializationContext jsondeserializationcontext) throws JsonParseException {
             return this.deserialize(jsonelement, type, jsondeserializationcontext);
         }
+         */ // Loom end
     }
+
+    // Loom start
+    @Override
+    public Object clone() throws CloneNotSupportedException {
+        return super.clone();
+    }
+    // Loom end
 }
