--- a/net/minecraft/server/command/CommandOutput.java
+++ b/net/minecraft/server/command/CommandOutput.java
@@ -19,6 +19,13 @@
         public boolean shouldBroadcastConsoleToOps() {
             return false;
         }
+
+        // Loom start :: implement command source
+        @Override
+        public org.loomdev.api.command.CommandSource getCommandSource(ServerCommandSource source) {
+            throw new UnsupportedOperationException("Not implemented.");
+        }
+        // Loom
     };
 
     void sendSystemMessage(Text arg, UUID uUID);
@@ -28,4 +35,6 @@
     boolean shouldTrackOutput();
 
     boolean shouldBroadcastConsoleToOps();
+
+    org.loomdev.api.command.CommandSource getCommandSource(ServerCommandSource source); // Loom
 }
