--- a/net/minecraft/resource/ServerResourceManager.java
+++ b/net/minecraft/resource/ServerResourceManager.java
@@ -17,7 +17,7 @@
 
     private static final CompletableFuture<Unit> field_25334 = CompletableFuture.completedFuture(Unit.INSTANCE);
     private final ReloadableResourceManager resourceManager;
-    private final CommandManager commandManager;
+    public final CommandManager commandManager; // Loom :: public access
     private final RecipeManager recipeManager;
     private final RegistryTagManager registryTagManager;
     private final LootConditionManager lootConditionManager;
@@ -40,6 +40,11 @@
         this.resourceManager.registerListener(this.lootManager);
         this.resourceManager.registerListener(this.functionLoader);
         this.resourceManager.registerListener(this.serverAdvancementLoader);
+        // Loom start :: register plugin commands
+        if (org.loomdev.api.Loom.getServer() != null) {
+            ((org.loomdev.loom.command.CommandManagerImpl) org.loomdev.api.Loom.getCommandManager()).internalReload();
+        }
+        // Loom end
     }
 
     public FunctionLoader getFunctionLoader() {
