--- a/net/minecraft/entity/mob/EndermiteEntity.java
+++ b/net/minecraft/entity/mob/EndermiteEntity.java
@@ -28,7 +28,12 @@
 
 public class EndermiteEntity extends HostileEntity {
 
-    private int lifeTime;
+    // Loom start
+    public java.util.OptionalInt overrideMaxLifeTime = java.util.OptionalInt.empty(); // Max life time override
+    public int defaultMaxLifeTime = 2400;
+    // Loom end
+
+    public int lifeTime; // Loom :: public accessor
     private boolean playerSpawned;
 
     public EndermiteEntity(EntityType<? extends EndermiteEntity> entitytype, World world) {
@@ -119,7 +124,7 @@
                 ++this.lifeTime;
             }
 
-            if (this.lifeTime >= 2400) {
+            if (this.lifeTime >= overrideMaxLifeTime.orElse(defaultMaxLifeTime)) { // Loom :: Max life time override
                 this.method_31472();
             }
         }
