--- a/net/minecraft/entity/mob/PiglinEntity.java
+++ b/net/minecraft/entity/mob/PiglinEntity.java
@@ -52,9 +52,9 @@
 
 public class PiglinEntity extends AbstractPiglinEntity implements CrossbowUser {
 
-    private static final TrackedData<Boolean> BABY = DataTracker.registerData(PiglinEntity.class, TrackedDataHandlerRegistry.BOOLEAN);
-    private static final TrackedData<Boolean> CHARGING = DataTracker.registerData(PiglinEntity.class, TrackedDataHandlerRegistry.BOOLEAN);
-    private static final TrackedData<Boolean> DANCING = DataTracker.registerData(PiglinEntity.class, TrackedDataHandlerRegistry.BOOLEAN);
+    public static final TrackedData<Boolean> BABY = DataTracker.registerData(PiglinEntity.class, TrackedDataHandlerRegistry.BOOLEAN); // Loom :: public accessor
+    public static final TrackedData<Boolean> CHARGING = DataTracker.registerData(PiglinEntity.class, TrackedDataHandlerRegistry.BOOLEAN); // Loom :: public accessor
+    public static final TrackedData<Boolean> DANCING = DataTracker.registerData(PiglinEntity.class, TrackedDataHandlerRegistry.BOOLEAN); // Loom :: public accessor
     private static final UUID BABY_SPEED_BOOST_ID = UUID.fromString("766bfa64-11f3-11ea-8d71-362b9e155667");
     private static final EntityAttributeModifier BABY_SPEED_BOOST = new EntityAttributeModifier(PiglinEntity.BABY_SPEED_BOOST_ID, "Baby speed boost", 0.20000000298023224D, EntityAttributeModifier.Operation.MULTIPLY_BASE);
     private final SimpleInventory inventory = new SimpleInventory(8);
@@ -173,7 +173,7 @@
     }
 
     public Brain<PiglinEntity> getBrain() {
-        return super.getBrain();
+        return (Brain<PiglinEntity>) super.getBrain(); // Loom :: decompile error
     }
 
     public ActionResult interactMob(PlayerEntity playerentity, Hand hand) {
@@ -351,7 +351,7 @@
     }
 
     protected SoundEvent getAmbientSound() {
-        return this.world.isClient ? null : (SoundEvent) PiglinBrain.method_30091(this).orElse((Object) null);
+        return this.world.isClient ? null : (SoundEvent) PiglinBrain.method_30091(this).orElse(null); // Loom :: decompile error
     }
 
     protected SoundEvent getHurtSound(DamageSource damagesource) {
