--- a/net/minecraft/entity/player/PlayerAbilities.java
+++ b/net/minecraft/entity/player/PlayerAbilities.java
@@ -9,8 +9,8 @@
     public boolean allowFlying;
     public boolean creativeMode;
     public boolean allowModifyWorld = true;
-    private float flySpeed = 0.05F;
-    private float walkSpeed = 0.1F;
+    public float flySpeed = 0.05F; // Loom :: public accessor
+    public float walkSpeed = 0.1F; // Loom :: public accessor
 
     public PlayerAbilities() {}
 
