--- a/net/minecraft//world/entity/animal/horse/AbstractHorse.java
+++ b/net/minecraft//world/entity/animal/horse/AbstractHorse.java
@@ -71,6 +71,7 @@
 
 public abstract class AbstractHorse extends Animal implements ContainerListener, PlayerRideableJumping, Saddleable {
 
+    public java.util.OptionalInt maxTemperOverride = java.util.OptionalInt.empty(); // Loom :: (AbstractHorseImpl)
     public static final int EQUIPMENT_SLOT_OFFSET = 400;
     public static final int CHEST_SLOT_OFFSET = 499;
     public static final int INVENTORY_SLOT_OFFSET = 500;
@@ -393,7 +394,7 @@
     }
 
     public int getMaxTemper() {
-        return 100;
+        return maxTemperOverride.orElse(100); // Loom
     }
 
     protected float getSoundVolume() {
