--- a/net/minecraft/item/ArmorStandItem.java
+++ b/net/minecraft/item/ArmorStandItem.java
@@ -49,6 +49,11 @@
                         return ActionResult.FAIL;
                     }
 
+                    // Loom start :: fire ArmorStandPlacedEvent
+                    if (itemplacementcontext.getPlayer() != null && org.loomdev.loom.event.LoomEventDispatcher.onItemEntityPlace(armorstandentity, itemplacementcontext.getPlayer()).isCancelled()) {
+                        return ActionResult.FAIL;
+                    } // TODO dispenser event
+                    // Loom end
                     serverworld.spawnEntityAndPassengers(armorstandentity);
                     float f = (float) MathHelper.floor((MathHelper.wrapDegrees(itemusagecontext.getPlayerYaw() - 180.0F) + 22.5F) / 45.0F) * 45.0F;
 
