--- a/net/minecraft/item/BlockItem.java
+++ b/net/minecraft/item/BlockItem.java
@@ -65,6 +65,13 @@
                         blockstate1 = this.placeFromTag(blockpos, world, itemstack, blockstate1);
                         this.postPlacement(blockpos, world, playerentity, itemstack, blockstate1);
                         block.onPlaced(world, blockpos, blockstate1, playerentity, itemstack);
+
+                        // Loom start :: fire BlockPlacedEvent
+                        if (playerentity != null && org.loomdev.loom.event.LoomEventDispatcher.onEntityBlockPlace(world, blockpos, playerentity).isCancelled()) {
+                            return ActionResult.FAIL;
+                        }
+                        // Loom end
+
                         if (playerentity instanceof ServerPlayerEntity) {
                             Criteria.PLACED_BLOCK.trigger((ServerPlayerEntity) playerentity, blockpos, itemstack);
                         }
