--- a/net/minecraft/item/FireChargeItem.java
+++ b/net/minecraft/item/FireChargeItem.java
@@ -9,6 +9,7 @@
 import net.minecraft.util.ActionResult;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
+import org.loomdev.api.event.block.BlockIgniteEvent;
 
 public class FireChargeItem extends Item {
 
@@ -23,12 +24,22 @@
         boolean flag = false;
 
         if (CampfireBlock.method_30035(blockstate)) {
+            // Loom start :: fire BlockIgnitedEvent
+            if (itemusagecontext.getPlayer() != null && org.loomdev.loom.event.LoomEventDispatcher.onBlockIgnite(world, blockpos, itemusagecontext.getPlayer(), BlockIgniteEvent.Cause.FIRE_CHARGE).isCancelled()) {
+                return ActionResult.FAIL;
+            }
+            // Loom end
             this.playUseSound(world, blockpos);
             world.setBlockState(blockpos, (BlockState) blockstate.with(CampfireBlock.LIT, true));
             flag = true;
         } else {
             blockpos = blockpos.offset(itemusagecontext.getSide());
             if (AbstractFireBlock.method_30032(world, blockpos, itemusagecontext.getPlayerFacing())) {
+                // Loom start :: fire BlockIgnitedEvent
+                if (itemusagecontext.getPlayer() != null && org.loomdev.loom.event.LoomEventDispatcher.onBlockIgnite(world, blockpos, itemusagecontext.getPlayer(), BlockIgniteEvent.Cause.FIRE_CHARGE).isCancelled()) {
+                    return ActionResult.FAIL;
+                }
+                // Loom end
                 this.playUseSound(world, blockpos);
                 world.setBlockState(blockpos, AbstractFireBlock.getState(world, blockpos));
                 flag = true;
