--- a/net/minecraft/block/BedBlock.java
+++ b/net/minecraft/block/BedBlock.java
@@ -39,6 +39,7 @@
 import net.minecraft.world.explosion.Explosion;
 import net.minecraft.world.explosion.ExplosionBehavior;
 import org.apache.commons.lang3.ArrayUtils;
+import org.loomdev.api.event.entity.movement.EntityBounceEvent;
 
 public class BedBlock extends HorizontalFacingBlock implements BlockEntityProvider {
 
@@ -135,7 +136,17 @@
         if (vec3d.y < 0.0D) {
             double d0 = entity instanceof LivingEntity ? 1.0D : 0.8D;
 
-            entity.setVelocity(vec3d.x, -vec3d.y * 0.6600000262260437D * d0, vec3d.z);
+            // Loom :: fire EntityBouncedEvent
+            EntityBounceEvent event = org.loomdev.loom.event.LoomEventDispatcher.onEntityBounce(entity, d0);
+
+            if (event.isCancelled()) {
+                entity.setVelocity(vec3d.x, 0, vec3d.z);
+            } else {
+                entity.setVelocity(vec3d.x, -vec3d.y * 0.6600000262260437D * event.getBounceMultiplier(), vec3d.z);
+            }
+
+            entity.velocityModified = true;
+            // Loom end
         }
 
     }
