--- a/net/minecraft/block/Block.java
+++ b/net/minecraft/block/Block.java
@@ -57,9 +57,7 @@
             return !VoxelShapes.matchesAnywhere(VoxelShapes.fullCube(), voxelshape, BooleanBiFunction.NOT_SAME);
         }
 
-        public Object load(Object object) throws Exception {
-            return this.load((VoxelShape) object);
-        }
+        // Loom :: decompile error
     });
     protected final StateManager<Block, BlockState> stateManager;
     private BlockState defaultState;
@@ -256,6 +254,10 @@
 
     protected void dropExperience(ServerWorld serverworld, BlockPos blockpos, int i) {
         if (serverworld.getGameRules().getBoolean(GameRules.DO_TILE_DROPS)) {
+            // Loom start :: fire BlockDroppedExperienceEvent
+            org.loomdev.api.event.block.BlockDropExperienceEvent event = org.loomdev.loom.event.LoomEventDispatcher.onBlockDropExperience(serverworld, blockpos, i);
+            if (event.isCancelled()) return;
+            // Loom end
             ExperienceOrbEntity.method_31493(serverworld, Vec3d.ofCenter(blockpos), i);
         }
 
