--- a/net/minecraft/block/CoralFanBlock.java
+++ b/net/minecraft/block/CoralFanBlock.java
@@ -23,6 +23,11 @@
 
     public void scheduledTick(BlockState blockstate, ServerWorld serverworld, BlockPos blockpos, Random random) {
         if (!isInWater(blockstate, serverworld, blockpos)) {
+            // Loom start :: fire CoralDiedEvent
+            if (org.loomdev.loom.event.LoomEventDispatcher.onPlantDie(serverworld, blockpos).isCancelled()) {
+                return;
+            }
+            // Loom end
             serverworld.setBlockState(blockpos, (BlockState) this.deadCoralBlock.getDefaultState().with(CoralFanBlock.WATERLOGGED, false), 2);
         }
 
