--- a/net/minecraft/block/WetSpongeBlock.java
+++ b/net/minecraft/block/WetSpongeBlock.java
@@ -14,6 +14,11 @@
 
     public void onBlockAdded(BlockState blockstate, World world, BlockPos blockpos, BlockState blockstate1, boolean flag) {
         if (world.getDimension().isUltrawarm()) {
+            // Loom start :: fire BlockEvaporatedEvent
+            if (org.loomdev.loom.event.LoomEventDispatcher.onBlockEvaporate(world, blockpos).isCancelled()) {
+                return;
+            }
+            // Loom end
             world.setBlockState(blockpos, Blocks.SPONGE.getDefaultState(), 3);
             world.syncWorldEvent(2009, blockpos, 0);
             world.playSound((PlayerEntity) null, blockpos, SoundEvents.BLOCK_FIRE_EXTINGUISH, SoundCategory.BLOCKS, 1.0F, (1.0F + world.getRandom().nextFloat() * 0.2F) * 0.7F);
