--- a/net/minecraft/entity/passive/AbstractTraderEntity.java
+++ b/net/minecraft/entity/passive/AbstractTraderEntity.java
@@ -39,7 +39,7 @@
     @Nullable
     private PlayerEntity customer;
     @Nullable
-    protected TraderOfferList offers;
+    public TraderOfferList offers; // Loom :: public accessor
     private final SimpleInventory inventory = new SimpleInventory(8);
 
     public AbstractTraderEntity(EntityType<? extends AbstractTraderEntity> entitytype, World world) {
