--- a/net/minecraft/entity/passive/BeeEntity.java
+++ b/net/minecraft/entity/passive/BeeEntity.java
@@ -98,10 +98,10 @@
     private float lastPitch;
     private int ticksSinceSting;
     private int ticksSincePollination;
-    private int cannotEnterHiveTicks;
+    public int cannotEnterHiveTicks; // Loom :: public accessor
     private int cropsGrownSincePollination;
     private int ticksLeftToFindHive;
-    private int ticksUntilCanPollinate;
+    public int ticksUntilCanPollinate; // Loom :: public accessor
     @Nullable
     private BlockPos flowerPos;
     @Nullable
@@ -440,7 +440,7 @@
         return this.getBeeFlag(8);
     }
 
-    private void setHasNectar(boolean flag) {
+    public void setHasNectar(boolean flag) { // Loom :: public accessor
         if (flag) {
             this.resetPollinationTicks();
         }
@@ -452,7 +452,7 @@
         return this.getBeeFlag(4);
     }
 
-    private void setHasStung(boolean flag) {
+    public void setHasStung(boolean flag) { // Loom :: public accessor
         this.setBeeFlag(4, flag);
     }
 
@@ -544,7 +544,7 @@
 
     protected void fall(double d0, boolean flag, BlockState blockstate, BlockPos blockpos) {}
 
-    protected boolean hasWings() {
+    public boolean hasWings() { // Loom :: public accessor (inherited from Entity)
         return true;
     }
 
@@ -579,9 +579,7 @@
         return blockpos.isWithinDistance((Vec3i) this.getBlockPos(), (double) i);
     }
 
-    public PassiveEntity createChild(ServerWorld serverworld, PassiveEntity passiveentity) {
-        return this.createChild(serverworld, passiveentity);
-    }
+    // Loom :: decompile error
 
     class EnterHiveGoal extends BeeEntity.NotAngryGoal {
 
