--- a/net/minecraft/entity/passive/FoxEntity.java
+++ b/net/minecraft/entity/passive/FoxEntity.java
@@ -311,15 +311,15 @@
         return FoxEntity.Type.fromId((Integer) this.dataTracker.get(FoxEntity.TYPE));
     }
 
-    private void setType(FoxEntity.Type foxentity_type) {
+    public void setType(FoxEntity.Type foxentity_type) { // Loom :: public accessor
         this.dataTracker.set(FoxEntity.TYPE, foxentity_type.getId());
     }
 
-    private List<UUID> getTrustedUuids() {
+    public List<UUID> getTrustedUuids() { // Loom :: public accessor
         List<UUID> list = Lists.newArrayList();
 
-        list.add(((Optional) this.dataTracker.get(FoxEntity.OWNER)).orElse((Object) null));
-        list.add(((Optional) this.dataTracker.get(FoxEntity.OTHER_TRUSTED)).orElse((Object) null));
+        list.add((this.dataTracker.get(FoxEntity.OWNER)).orElse(null)); // Loom :: decompile error
+        list.add((this.dataTracker.get(FoxEntity.OTHER_TRUSTED)).orElse(null)); // Loom :: decompile error
         return list;
     }
 
@@ -399,7 +399,7 @@
         return this.getFoxFlag(32);
     }
 
-    private void setSleeping(boolean flag) {
+    public void setSleeping(boolean flag) { // Loom :: public accessor
         this.setFoxFlag(32, flag);
     }
 
@@ -645,9 +645,7 @@
         return true;
     }
 
-    public PassiveEntity createChild(ServerWorld serverworld, PassiveEntity passiveentity) {
-        return this.createChild(serverworld, passiveentity);
-    }
+    // Loom :: decompile error
 
     class LookAtEntityGoal extends net.minecraft.entity.ai.goal.LookAtEntityGoal {
 
@@ -1090,9 +1088,7 @@
             return livingentity instanceof FoxEntity ? false : (!(livingentity instanceof ChickenEntity) && !(livingentity instanceof RabbitEntity) && !(livingentity instanceof HostileEntity) ? (livingentity instanceof TameableEntity ? !((TameableEntity) livingentity).isTamed() : (livingentity instanceof PlayerEntity && (livingentity.isSpectator() || ((PlayerEntity) livingentity).isCreative()) ? false : (FoxEntity.this.canTrust(livingentity.getUuid()) ? false : !livingentity.isSleeping() && !livingentity.isSneaky()))) : true);
         }
 
-        public boolean test(Object object) {
-            return this.test((LivingEntity) object);
-        }
+        // Loom :: decompile error
     }
 
     class AvoidDaylightGoal extends EscapeSunlightGoal {
