--- a/net/minecraft/entity/passive/MerchantEntity.java
+++ b/net/minecraft/entity/passive/MerchantEntity.java
@@ -39,7 +39,7 @@
     @Nullable
     private PlayerEntity customer;
     @Nullable
-    protected TradeOfferList offers;
+    public TradeOfferList offers; // Loom :: public accessor
     private final SimpleInventory inventory = new SimpleInventory(8);
 
     public MerchantEntity(EntityType<? extends MerchantEntity> entitytype, World world) {
