--- a/net/minecraft/entity/passive/MooshroomEntity.java
+++ b/net/minecraft/entity/passive/MooshroomEntity.java
@@ -41,9 +41,9 @@
 
 public class MooshroomEntity extends CowEntity implements Shearable {
 
-    private static final TrackedData<String> TYPE = DataTracker.registerData(MooshroomEntity.class, TrackedDataHandlerRegistry.STRING);
-    private StatusEffect stewEffect;
-    private int stewEffectDuration;
+    public static final TrackedData<String> TYPE = DataTracker.registerData(MooshroomEntity.class, TrackedDataHandlerRegistry.STRING); // Loom :: public accessor
+    public StatusEffect stewEffect; // Loom :: public accessor
+    public int stewEffectDuration; // Loom :: public accessor
     private UUID lightningId;
 
     public MooshroomEntity(EntityType<? extends MooshroomEntity> entitytype, World world) {
@@ -247,19 +247,13 @@
         return mooshroomentity_type2;
     }
 
-    public CowEntity createChild(ServerWorld serverworld, PassiveEntity passiveentity) {
-        return this.createChild(serverworld, passiveentity);
-    }
-
-    public PassiveEntity createChild(ServerWorld serverworld, PassiveEntity passiveentity) {
-        return this.createChild(serverworld, passiveentity);
-    }
+    // Loom :: decompile error
 
     public static enum Type {
 
         RED("red", Blocks.RED_MUSHROOM.getDefaultState()), BROWN("brown", Blocks.BROWN_MUSHROOM.getDefaultState());
 
-        private final String name;
+        public final String name; // Loom :: public accessor
         private final BlockState mushroom;
 
         private Type(String s, BlockState blockstate) {
