--- a/net/minecraft//world/entity/monster/Blaze.java
+++ b/net/minecraft//world/entity/monster/Blaze.java
@@ -95,7 +95,7 @@
     }
 
     public boolean isSensitiveToWater() {
-        return true;
+        return sensitiveToWaterOverride.orElse(true); // Loom :: (LivingEntityImpl)
     }
 
     protected void customServerAiStep() {
@@ -125,11 +125,11 @@
         return this.isCharged();
     }
 
-    private boolean isCharged() {
+    public boolean isCharged() { // Loom :: private -> public (BlazeImpl)
         return ((Byte) this.entityData.get(Blaze.DATA_FLAGS_ID) & 1) != 0;
     }
 
-    void setCharged(boolean flag) {
+    public void setCharged(boolean flag) { // Loom :: package-private -> public (BlazeImpl)
         byte b0 = (Byte) this.entityData.get(Blaze.DATA_FLAGS_ID);
 
         if (flag) {
