--- a/net/minecraft/world/entity/monster/Endermite.java
+++ b/net/minecraft/world/entity/monster/Endermite.java
@@ -28,7 +28,9 @@
 
 public class Endermite extends Monster {
 
-    private int life;
+    public int life; // Loom :: private -> public (EndermiteImpl)
+    public java.util.OptionalInt lifeTimeOverride = java.util.OptionalInt.empty(); // Loom :: (EndermiteImpl)
+    public int vanillaLife = 2400; // Loom :: (EndermiteImpl)
 
     public Endermite(EntityType<? extends Endermite> entitytype, Level level) {
         super(entitytype, level);
@@ -108,7 +110,7 @@
                 ++this.life;
             }
 
-            if (this.life >= 2400) {
+            if (this.life >= lifeTimeOverride.orElse(vanillaLife)) { // Loom :: (EndermiteImpl)
                 this.discard();
             }
         }
