--- a/net/minecraft/world/entity/monster/Guardian.java
+++ b/net/minecraft/world/entity/monster/Guardian.java
@@ -108,7 +108,7 @@
         return (Boolean) this.entityData.get(Guardian.DATA_ID_MOVING);
     }
 
-    private void setMoving(boolean flag) {
+    public void setMoving(boolean flag) { // Loom :: private -> public (GuardianImpl)
         this.entityData.set(Guardian.DATA_ID_MOVING, flag);
     }
 
@@ -116,7 +116,7 @@
         return 80;
     }
 
-    private void setActiveAttackTarget(int i) {
+    public void setActiveAttackTarget(int i) { // Loom :: private -> public (GuardianImpl)
         this.entityData.set(Guardian.DATA_ID_ATTACK_TARGET, i);
     }
 
@@ -456,8 +456,6 @@
             return (livingentity instanceof Player || livingentity instanceof Squid || livingentity != null && Axolotl.NOT_PLAYING_DEAD_SELECTOR.test(livingentity)) && livingentity.distanceToSqr((Entity) this.guardian) > 9.0D;
         }
 
-        public boolean test(@Nullable Object object) {
-            return this.test((LivingEntity) object);
-        }
+        // Loom :: decompile error
     }
 }
