--- a/net/minecraft/world/entity/monster/PatrollingMonster.java
+++ b/net/minecraft/world/entity/monster/PatrollingMonster.java
@@ -28,7 +28,7 @@
 
     private BlockPos patrolTarget;
     private boolean patrolLeader;
-    private boolean patrolling;
+    public boolean patrolling; // Loom :: private -> public
 
     protected PatrollingMonster(EntityType<? extends PatrollingMonster> entitytype, Level level) {
         super(entitytype, level);
