--- a/net/minecraft//world/entity/monster/Slime.java
+++ b/net/minecraft//world/entity/monster/Slime.java
@@ -47,7 +47,7 @@
 
 public class Slime extends Mob implements Enemy {
 
-    private static final EntityDataAccessor<Integer> ID_SIZE = SynchedEntityData.defineId(Slime.class, EntityDataSerializers.INT);
+    public static final EntityDataAccessor<Integer> ID_SIZE = SynchedEntityData.defineId(Slime.class, EntityDataSerializers.INT); // Loom :: private -> public
     public static final int MIN_SIZE = 1;
     public static final int MAX_SIZE = 127;
     public float targetSquish;
@@ -176,9 +176,11 @@
         super.onSyncedDataUpdated(entitydataaccessor);
     }
 
+    /* // Loom start :: decompile error
     public EntityType<? extends Slime> getType() {
         return super.getType();
     }
+     */ // Loom
 
     public void remove(Entity.RemovalReason entity_removalreason) {
         int i = this.getSize();
