--- a/net/minecraft//world/entity/monster/Strider.java
+++ b/net/minecraft//world/entity/monster/Strider.java
@@ -349,7 +349,7 @@
     }
 
     public boolean isSensitiveToWater() {
-        return true;
+        return sensitiveToWaterOverride.orElse(true); // Loom
     }
 
     public boolean isOnFire() {
@@ -443,9 +443,11 @@
         return new AgeableMob.AgeableMobGroupData(0.0F);
     }
 
+    /* // Loom start :: decompile error
     public AgeableMob getBreedOffspring(ServerLevel serverlevel, AgeableMob ageablemob) {
         return this.getBreedOffspring(serverlevel, ageablemob);
     }
+     */ // Loom end
 
     static class StriderGoToLavaGoal extends MoveToBlockGoal {
 
