--- a/net/minecraft//world/entity/monster/ZombieVillager.java
+++ b/net/minecraft//world/entity/monster/ZombieVillager.java
@@ -56,10 +56,10 @@
     private static final int MAX_SPECIAL_BLOCKS_COUNT = 14;
     private static final int SPECIAL_BLOCK_RADIUS = 4;
     private int villagerConversionTime;
-    private UUID conversionStarter;
+    public UUID conversionStarter; // Loom :: private -> public (ZombieVillagerImpl)
     private Tag gossips;
     private CompoundTag tradeOffers;
-    private int villagerXp;
+    public int villagerXp; // Loom :: private -> public (ZombieVillagerImpl)
 
     public ZombieVillager(EntityType<? extends ZombieVillager> entitytype, Level level) {
         super(entitytype, level);
@@ -74,7 +74,7 @@
 
     public void addAdditionalSaveData(CompoundTag compoundtag) {
         super.addAdditionalSaveData(compoundtag);
-        DataResult dataresult = VillagerData.CODEC.encodeStart(NbtOps.INSTANCE, this.getVillagerData());
+        DataResult<Tag> dataresult = VillagerData.CODEC.encodeStart(NbtOps.INSTANCE, this.getVillagerData()); // Loom :: decompile error
         Logger logger = ZombieVillager.LOGGER;
 
         Objects.requireNonNull(logger);
