--- a/net/minecraft/server/PlayerManager.java
+++ b/net/minecraft/server/PlayerManager.java
@@ -193,7 +193,9 @@
             translatabletext = new TranslatableText("multiplayer.player.joined.renamed", new Object[]{serverplayerentity.getDisplayName(), s});
         }
 
-        this.broadcastChatMessage(translatabletext.formatted(Formatting.YELLOW), MessageType.SYSTEM, Util.NIL_UUID);
+        org.loomdev.api.event.player.connection.PlayerJoinedEvent event = org.loomdev.loom.event.LoomEventDispatcher.onPlayerJoined(serverplayerentity, translatabletext.formatted(Formatting.YELLOW));
+        
+        this.broadcastChatMessage(org.loomdev.loom.util.transformer.TextTransformer.toMinecraft(event.getJoinMessage()), MessageType.SYSTEM, Util.NIL_UUID);
         serverplaynetworkhandler.requestTeleport(serverplayerentity.getX(), serverplayerentity.getY(), serverplayerentity.getZ(), serverplayerentity.yaw, serverplayerentity.pitch);
         this.players.add(serverplayerentity);
         this.playerMap.put(serverplayerentity.getUuid(), serverplayerentity);
@@ -637,7 +639,7 @@
     }
 
     public void removeFromOperators(GameProfile gameprofile) {
-        this.ops.remove((Object) gameprofile);
+        this.ops.remove(gameprofile); // Loom :: decompile error
         ServerPlayerEntity serverplayerentity = this.getPlayer(gameprofile.getId());
 
         if (serverplayerentity != null) {
