--- a/net/minecraft/server/world/ServerWorld.java
+++ b/net/minecraft/server/world/ServerWorld.java
@@ -150,7 +150,7 @@
     private final List<ServerPlayerEntity> players = Lists.newArrayList();
     private final ServerChunkManager serverChunkManager;
     private final MinecraftServer server;
-    private final ServerWorldProperties worldProperties;
+    public final ServerWorldProperties worldProperties; // Loom :: public access
     private final class_5574 field_26934 = new class_5574();
     private final class_5579<Entity> field_26935;
     public boolean savingDisabled;
@@ -216,6 +216,7 @@
             this.enderDragonFight = null;
         }
 
+        ((org.loomdev.loom.server.ServerImpl) org.loomdev.api.Loom.getServer()).registerWorld(this); // Loom
     }
 
     public void setWeather(int i, int j, boolean flag, boolean flag1) {
@@ -438,7 +439,7 @@
 
     private void wakeSleepingPlayers() {
         ((List) this.players.stream().filter(LivingEntity::isSleeping).collect(Collectors.toList())).forEach((serverplayerentity) -> {
-            serverplayerentity.wakeUp(false, false);
+            ((ServerPlayerEntity) serverplayerentity).wakeUp(false, false); // Loom :: decompile error
         });
     }
 
@@ -817,7 +818,7 @@
     }
 
     public boolean shouldCreateNewEntityWithPassenger(Entity entity) {
-        Stream stream = entity.streamPassengersRecursively().map(Entity::getUuid);
+        Stream<UUID> stream = entity.streamPassengersRecursively().map(Entity::getUuid); // Loom :: decompile error
         class_5579 class_5579 = this.field_26935;
 
         this.field_26935.getClass();
@@ -905,7 +906,7 @@
 
         explosion.collectBlocksAndDamageEntities();
         explosion.affectWorld(false);
-        if (explosion_destructiontype == Explosion.DestructionType.NONE) {
+        if (explosion.destructionType == Explosion.DestructionType.NONE) {
             explosion.clearAffectedBlocks();
         }
 
@@ -1426,8 +1427,8 @@
 
     private static <T> String method_31270(Iterable<T> iterable, Function<T, String> function) {
         try {
-            Object2IntOpenHashMap<String> object2intopenhashmap = new Object2IntOpenHashMap();
-            Iterator iterator = iterable.iterator();
+            Object2IntOpenHashMap<String> object2intopenhashmap = new Object2IntOpenHashMap<>();
+            Iterator<T> iterator = iterable.iterator(); // Loom :: decompile error
 
             while (iterator.hasNext()) {
                 T t0 = iterator.next();
@@ -1436,7 +1437,7 @@
                 object2intopenhashmap.addTo(s, 1);
             }
 
-            return (String) object2intopenhashmap.object2IntEntrySet().stream().sorted(Comparator.comparing(Entry::getIntValue).reversed()).limit(5L).map((entry) -> {
+            return (String) object2intopenhashmap.object2IntEntrySet().stream().sorted(Comparator.comparing(it.unimi.dsi.fastutil.objects.Object2IntMap.Entry::getIntValue)).sorted(java.util.Collections.reverseOrder()).limit(5L).map((entry) -> { // Loom :: decompile error
                 return (String) entry.getKey() + ":" + entry.getIntValue();
             }).collect(Collectors.joining(","));
         } catch (Exception exception) {
@@ -1479,21 +1480,7 @@
         return "Chunks[S] W: " + this.serverChunkManager.getDebugString() + " E: " + this.field_26935.method_31845();
     }
 
-    public Scoreboard getScoreboard() {
-        return this.getScoreboard();
-    }
-
-    public ChunkManager getChunkManager() {
-        return this.getChunkManager();
-    }
-
-    public TickScheduler getFluidTickScheduler() {
-        return this.getFluidTickScheduler();
-    }
-
-    public TickScheduler getBlockTickScheduler() {
-        return this.getBlockTickScheduler();
-    }
+    // Loom :: decompile error
 
     final class class_5526 implements class_5576<Entity> {
 
@@ -1563,32 +1550,6 @@
 
         }
 
-        public void method_31797(Object object) {
-            this.method_31797((Entity) object);
-        }
-
-        public void method_31798(Object object) {
-            this.method_31798((Entity) object);
-        }
-
-        public void method_31799(Object object) {
-            this.method_31799((Entity) object);
-        }
-
-        public void method_31800(Object object) {
-            this.method_31800((Entity) object);
-        }
-
-        public void method_31801(Object object) {
-            this.method_31801((Entity) object);
-        }
-
-        public void method_31802(Object object) {
-            this.method_31802((Entity) object);
-        }
-
-        class_5526(Object object) {
-            this();
-        }
+        // Loom :: decompile error
     }
 }
