--- a/net/minecraft/server/level/ServerLevel.java
+++ b/net/minecraft/server/level/ServerLevel.java
@@ -151,7 +151,7 @@
     private final List<ServerPlayer> players = Lists.newArrayList();
     private final ServerChunkCache chunkSource;
     private final MinecraftServer server;
-    private final ServerLevelData serverLevelData;
+    public final ServerLevelData serverLevelData; // Loom :: private -> public (WorldImpl)
     private final EntityTickList entityTickList = new EntityTickList();
     private final PersistentEntitySectionManager<Entity> entityManager;
     public boolean noSave;
@@ -440,7 +440,7 @@
     }
 
     private void wakeUpAllPlayers() {
-        ((List) this.players.stream().filter(LivingEntity::isSleeping).collect(Collectors.toList())).forEach((serverplayer) -> {
+        (this.players.stream().filter(LivingEntity::isSleeping).collect(Collectors.toList())).forEach((serverplayer) -> { // Loom :: decompile error
             serverplayer.stopSleepInBed(false, false);
         });
     }
@@ -817,7 +817,7 @@
     }
 
     public boolean tryAddFreshEntityWithPassengers(Entity entity) {
-        Stream stream = entity.getSelfAndPassengers().map(Entity::getUUID);
+        Stream<UUID> stream = entity.getSelfAndPassengers().map(Entity::getUUID); // Loom :: decompile error
         PersistentEntitySectionManager persistententitysectionmanager = this.entityManager;
 
         this.entityManager.getClass();
@@ -905,7 +905,7 @@
 
         explosion.explode();
         explosion.finalizeExplosion(false);
-        if (explosion_blockinteraction == Explosion.BlockInteraction.NONE) {
+        if (explosion.blockInteraction == Explosion.BlockInteraction.NONE) { // Loom
             explosion.clearToBlow();
         }
 
@@ -1425,7 +1425,7 @@
     private static <T> String getTypeCount(Iterable<T> iterable, Function<T, String> function) {
         try {
             Object2IntOpenHashMap<String> object2intopenhashmap = new Object2IntOpenHashMap();
-            Iterator iterator = iterable.iterator();
+            Iterator<T> iterator = iterable.iterator(); // Loom :: decompile error
 
             while (iterator.hasNext()) {
                 T t0 = iterator.next();
@@ -1434,7 +1434,7 @@
                 object2intopenhashmap.addTo(s, 1);
             }
 
-            return (String) object2intopenhashmap.object2IntEntrySet().stream().sorted(Comparator.comparing(Entry::getIntValue).reversed()).limit(5L).map((entry) -> {
+            return (String) object2intopenhashmap.object2IntEntrySet().stream().sorted(Comparator.comparing(Entry::getIntValue)).sorted(java.util.Collections.reverseOrder()).limit(5L).map((entry) -> { // Loom :: decompile error
                 return (String) entry.getKey() + ":" + entry.getIntValue();
             }).collect(Collectors.joining(","));
         } catch (Exception exception) {
@@ -1477,6 +1477,7 @@
         return "Chunks[S] W: " + this.chunkSource.gatherStats() + " E: " + this.entityManager.gatherStats();
     }
 
+    /* // Loom start :: decompile error
     public Scoreboard getScoreboard() {
         return this.getScoreboard();
     }
@@ -1492,6 +1493,7 @@
     public TickList getBlockTicks() {
         return this.getBlockTicks();
     }
+     */ // Loom end
 
     final class EntityCallbacks implements LevelCallback<Entity> {
 
@@ -1561,6 +1563,7 @@
 
         }
 
+        /* // Loom start :: decompile error
         public void onTrackingEnd(Object object) {
             this.onTrackingEnd((Entity) object);
         }
@@ -1588,5 +1591,6 @@
         EntityCallbacks(Object object) {
             this();
         }
+         */ // Loom end
     }
 }
