--- a/net/minecraft/server/level/ServerLevel.java
+++ b/net/minecraft/server/level/ServerLevel.java
@@ -146,12 +146,18 @@
 
 public class ServerLevel extends Level implements WorldGenLevel {
 
+    // Loom start
+    private final org.loomdev.loom.world.WorldImpl worldImpl;
+    public final org.loomdev.loom.world.WorldImpl getLoomWorld() {
+        return worldImpl;
+    }
+    // Loom end
     public static final BlockPos END_SPAWN_POINT = new BlockPos(100, 50, 0);
     private static final Logger LOGGER = LogManager.getLogger();
     private final List<ServerPlayer> players = Lists.newArrayList();
     private final ServerChunkCache chunkSource;
     private final MinecraftServer server;
-    private final ServerLevelData serverLevelData;
+    public final ServerLevelData serverLevelData; // Loom :: private -> public
     private final EntityTickList entityTickList = new EntityTickList();
     private final PersistentEntitySectionManager<Entity> entityManager;
     public boolean noSave;
@@ -217,6 +223,10 @@
             this.dragonFight = null;
         }
 
+        // Loom start
+        worldImpl = new org.loomdev.loom.world.WorldImpl(this);
+        ((org.loomdev.loom.server.ServerImpl) org.loomdev.api.Loom.getServer()).registerWorld(worldImpl);
+        // Loom end
     }
 
     public void setWeatherParameters(int i, int j, boolean flag, boolean flag1) {
@@ -438,7 +448,7 @@
     }
 
     private void wakeUpAllPlayers() {
-        ((List) this.players.stream().filter(LivingEntity::isSleeping).collect(Collectors.toList())).forEach((serverplayer) -> {
+        (this.players.stream().filter(LivingEntity::isSleeping).collect(Collectors.toList())).forEach((serverplayer) -> { // Loom :: decompile error
             serverplayer.stopSleepInBed(false, false);
         });
     }
@@ -818,7 +828,7 @@
     }
 
     public boolean tryAddFreshEntityWithPassengers(Entity entity) {
-        Stream stream = entity.getSelfAndPassengers().map(Entity::getUUID);
+        Stream<UUID> stream = entity.getSelfAndPassengers().map(Entity::getUUID); // Loom :: decompile error
         PersistentEntitySectionManager persistententitysectionmanager = this.entityManager;
 
         this.entityManager.getClass();
@@ -906,7 +916,7 @@
 
         explosion.explode();
         explosion.finalizeExplosion(false);
-        if (explosion_blockinteraction == Explosion.BlockInteraction.NONE) {
+        if (explosion.blockInteraction == Explosion.BlockInteraction.NONE) { // Loom
             explosion.clearToBlow();
         }
 
@@ -1428,7 +1438,7 @@
     private static <T> String getTypeCount(Iterable<T> iterable, Function<T, String> function) {
         try {
             Object2IntOpenHashMap<String> object2intopenhashmap = new Object2IntOpenHashMap();
-            Iterator iterator = iterable.iterator();
+            Iterator<T> iterator = iterable.iterator(); // Loom :: decompile error
 
             while (iterator.hasNext()) {
                 T t0 = iterator.next();
@@ -1437,7 +1447,7 @@
                 object2intopenhashmap.addTo(s, 1);
             }
 
-            return (String) object2intopenhashmap.object2IntEntrySet().stream().sorted(Comparator.comparing(Entry::getIntValue).reversed()).limit(5L).map((entry) -> {
+            return (String) object2intopenhashmap.object2IntEntrySet().stream().sorted(Comparator.comparing(Entry::getIntValue)).sorted(java.util.Collections.reverseOrder()).limit(5L).map((entry) -> { // Loom :: decompile error
                 return (String) entry.getKey() + ":" + entry.getIntValue();
             }).collect(Collectors.joining(","));
         } catch (Exception exception) {
@@ -1480,6 +1490,7 @@
         return "Chunks[S] W: " + this.chunkSource.gatherStats() + " E: " + this.entityManager.gatherStats();
     }
 
+    /* // Loom start :: decompile error
     public Scoreboard getScoreboard() {
         return this.getScoreboard();
     }
@@ -1495,6 +1506,7 @@
     public TickList getBlockTicks() {
         return this.getBlockTicks();
     }
+     */ // Loom end
 
     final class EntityCallbacks implements LevelCallback<Entity> {
 
@@ -1564,6 +1576,7 @@
 
         }
 
+        /* // Loom start :: decompile error
         public void onTrackingEnd(Object object) {
             this.onTrackingEnd((Entity) object);
         }
@@ -1591,5 +1604,6 @@
         EntityCallbacks(Object object) {
             this();
         }
+         */ // Loom end
     }
 }
