--- a/net/minecraft/entity/vehicle/BoatEntity.java
+++ b/net/minecraft/entity/vehicle/BoatEntity.java
@@ -380,7 +380,7 @@
         this.dataTracker.set(BoatEntity.RIGHT_PADDLE_MOVING, flag1);
     }
 
-    private BoatEntity.Location checkLocation() {
+    public BoatEntity.Location checkLocation() { // Loom :: public accessor
         BoatEntity.Location boatentity_location = this.getUnderWaterLocation();
 
         if (boatentity_location != null) {
