--- a/net/minecraft//sounds/SoundSource.java
+++ b/net/minecraft//sounds/SoundSource.java
@@ -1,5 +1,9 @@
 package net.minecraft.sounds;
 
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+
 public enum SoundSource {
 
     MASTER("master"), MUSIC("music"), RECORDS("record"), WEATHER("weather"), BLOCKS("block"), HOSTILE("hostile"), NEUTRAL("neutral"), PLAYERS("player"), AMBIENT("ambient"), VOICE("voice");
@@ -17,4 +21,11 @@
     private static SoundSource[] $values() {
         return new SoundSource[]{SoundSource.MASTER, SoundSource.MUSIC, SoundSource.RECORDS, SoundSource.WEATHER, SoundSource.BLOCKS, SoundSource.HOSTILE, SoundSource.NEUTRAL, SoundSource.PLAYERS, SoundSource.AMBIENT, SoundSource.VOICE};
     }
+
+    // Loom start
+    private static final Map<String, SoundSource> BY_NAME = Arrays.stream(SoundSource.values()).collect(Collectors.toMap(ss -> ss.name, ss -> ss));
+    public static SoundSource getByName(String name) {
+        return BY_NAME.get(name);
+    }
+    // Loom end
 }
