--- a/net/minecraft/server/network/ServerPlayerInteractionManager.java
+++ b/net/minecraft/server/network/ServerPlayerInteractionManager.java
@@ -236,6 +236,12 @@
     public boolean tryBreakBlock(BlockPos blockpos) {
         BlockState blockstate = this.world.getBlockState(blockpos);
 
+        // Loom start :: fire BlockBrokenEvent
+        if (org.loomdev.loom.event.LoomEventDispatcher.onBlockBroken(world, blockpos, player).isCancelled()) {
+            return false;
+        }
+        // Loom end
+
         if (!this.player.getMainHandStack().getItem().canMine(blockstate, this.world, blockpos, this.player)) {
             return false;
         } else {
