--- a/net/minecraft/entity/passive/AnimalEntity.java
+++ b/net/minecraft/entity/passive/AnimalEntity.java
@@ -28,7 +28,7 @@
 public abstract class AnimalEntity extends PassiveEntity {
 
     private int loveTicks;
-    private UUID lovingPlayer;
+    public UUID lovingPlayer; // Loom :: public accessor
 
     protected AnimalEntity(EntityType<? extends AnimalEntity> entitytype, World world) {
         super(entitytype, world);
