--- a/net/minecraft/entity/passive/HorseBaseEntity.java
+++ b/net/minecraft/entity/passive/HorseBaseEntity.java
@@ -68,6 +68,8 @@
 
 public abstract class HorseBaseEntity extends AnimalEntity implements InventoryChangedListener, JumpingMount, Saddleable {
 
+    public java.util.OptionalInt maxTemperOverride = java.util.OptionalInt.empty(); // Loom :: Max temper override
+
     private static final Predicate<LivingEntity> IS_BRED_HORSE = (livingentity) -> {
         return livingentity instanceof HorseBaseEntity && ((HorseBaseEntity) livingentity).isBred();
     };
@@ -378,7 +380,7 @@
     }
 
     public int getMaxTemper() {
-        return 100;
+        return maxTemperOverride.orElse(100); // Loom :: Max temper override
     }
 
     protected float getSoundVolume() {
