--- a/net/minecraft/entity/passive/LlamaEntity.java
+++ b/net/minecraft/entity/passive/LlamaEntity.java
@@ -53,9 +53,9 @@
 public class LlamaEntity extends AbstractDonkeyEntity implements RangedAttackMob {
 
     private static final Ingredient TAMING_INGREDIENT = Ingredient.ofItems(Items.WHEAT, Blocks.HAY_BLOCK.asItem());
-    private static final TrackedData<Integer> STRENGTH = DataTracker.registerData(LlamaEntity.class, TrackedDataHandlerRegistry.INTEGER);
+    public static final TrackedData<Integer> STRENGTH = DataTracker.registerData(LlamaEntity.class, TrackedDataHandlerRegistry.INTEGER); // Loom :: public accessor
     private static final TrackedData<Integer> CARPET_COLOR = DataTracker.registerData(LlamaEntity.class, TrackedDataHandlerRegistry.INTEGER);
-    private static final TrackedData<Integer> VARIANT = DataTracker.registerData(LlamaEntity.class, TrackedDataHandlerRegistry.INTEGER);
+    public static final TrackedData<Integer> VARIANT = DataTracker.registerData(LlamaEntity.class, TrackedDataHandlerRegistry.INTEGER); // Loom :: public accessor
     private boolean spit;
     @Nullable
     private LlamaEntity following;
@@ -332,7 +332,7 @@
     }
 
     public int getMaxTemper() {
-        return 30;
+        return maxTemperOverride.orElse(30); // Loom :: Max temper override
     }
 
     public boolean canBreedWith(AnimalEntity animalentity) {
@@ -448,9 +448,7 @@
         this.spitAt(livingentity);
     }
 
-    public PassiveEntity createChild(ServerWorld serverworld, PassiveEntity passiveentity) {
-        return this.createChild(serverworld, passiveentity);
-    }
+    // Loom :: decompile error
 
     static class ChaseWolvesGoal extends FollowTargetGoal<WolfEntity> {
 
