--- a/net/minecraft/entity/passive/LlamaEntity.java
+++ b/net/minecraft/entity/passive/LlamaEntity.java
@@ -332,7 +332,7 @@
     }
 
     public int getMaxTemper() {
-        return 30;
+        return maxTemperOverride.orElse(30); // Loom :: Max temper override
     }
 
     public boolean canBreedWith(AnimalEntity animalentity) {
@@ -448,9 +448,7 @@
         this.spitAt(livingentity);
     }
 
-    public PassiveEntity createChild(ServerWorld serverworld, PassiveEntity passiveentity) {
-        return this.createChild(serverworld, passiveentity);
-    }
+    // Loom :: decompile error
 
     static class ChaseWolvesGoal extends FollowTargetGoal<WolfEntity> {
 
