--- a/net/minecraft/entity/passive/PassiveEntity.java
+++ b/net/minecraft/entity/passive/PassiveEntity.java
@@ -16,6 +16,13 @@
 
 public abstract class PassiveEntity extends PathAwareEntity {
 
+    // Loom start :: getLoomEntity
+    @Override
+    public org.loomdev.loom.entity.passive.PassiveEntityImpl getLoomEntity() {
+        return (org.loomdev.loom.entity.passive.PassiveEntityImpl) super.getLoomEntity();
+    }
+    // Loom end :: getLoomEntity
+
     private static final TrackedData<Boolean> CHILD = DataTracker.registerData(PassiveEntity.class, TrackedDataHandlerRegistry.BOOLEAN);
     protected int breedingAge;
     protected int forcedAge;
