--- a/net/minecraft/entity/passive/StriderEntity.java
+++ b/net/minecraft/entity/passive/StriderEntity.java
@@ -349,7 +349,7 @@
     }
 
     public boolean hurtByWater() {
-        return true;
+        return hurtByWaterOverride.orElse(true); // hurt by water override
     }
 
     public boolean isOnFire() {
@@ -457,9 +457,7 @@
         return super.initialize(worldaccess, localdifficulty, spawnreason, (EntityData) entitydata, compoundtag);
     }
 
-    public PassiveEntity createChild(PassiveEntity passiveentity) {
-        return this.createChild(passiveentity);
-    }
+    // Loom :: decompile error
 
     static class Navigation extends MobNavigation {
 
