--- a/net/minecraft/entity/passive/StriderEntity.java
+++ b/net/minecraft/entity/passive/StriderEntity.java
@@ -343,7 +343,7 @@
     }
 
     public boolean hurtByWater() {
-        return true;
+        return hurtByWaterOverride.orElse(true); // hurt by water override
     }
 
     public boolean isOnFire() {
@@ -433,9 +433,7 @@
         return new PassiveEntity.PassiveData(0.0F);
     }
 
-    public PassiveEntity method_5613(ServerWorld serverworld, PassiveEntity passiveentity) {
-        return this.method_5613(serverworld, passiveentity);
-    }
+    // Loom :: decompile error
 
     static class class_5494 extends MoveToTargetPosGoal {
 
