--- a/net/minecraft/entity/passive/WanderingTraderEntity.java
+++ b/net/minecraft/entity/passive/WanderingTraderEntity.java
@@ -49,8 +49,8 @@
 public class WanderingTraderEntity extends AbstractTraderEntity {
 
     @Nullable
-    private BlockPos wanderTarget;
-    private int despawnDelay;
+    public BlockPos wanderTarget; // Loom :: public accessor
+    public int despawnDelay; // Loom :: public accessor
 
     public WanderingTraderEntity(EntityType<? extends WanderingTraderEntity> entitytype, World world) {
         super(entitytype, world);
