--- a/net/minecraft//world/entity/item/FallingBlockEntity.java
+++ b/net/minecraft//world/entity/item/FallingBlockEntity.java
@@ -46,7 +46,7 @@
 
 public class FallingBlockEntity extends Entity {
 
-    private BlockState blockState;
+    public BlockState blockState; // Loom :: private -> public
     public int time;
     public boolean dropItem;
     private boolean cancelDrop;
@@ -231,7 +231,7 @@
             if (i < 0) {
                 return false;
             } else {
-                Predicate predicate;
+                Predicate<Entity> predicate; // Loom :: decompile error
                 DamageSource damagesource1;
 
                 if (this.blockState.getBlock() instanceof Fallable) {
