--- a/net/minecraft/server/PlayerManager.java
+++ b/net/minecraft/server/PlayerManager.java
@@ -195,7 +195,13 @@
             translatabletext = new TranslatableText("multiplayer.player.joined.renamed", new Object[]{serverplayerentity.getDisplayName(), s});
         }
 
-        this.broadcastChatMessage(translatabletext.formatted(Formatting.YELLOW), MessageType.SYSTEM, Util.NIL_UUID);
+        // Loom start :: fire PlayerJoinedEvent
+        org.loomdev.loom.event.LoomEventDispatcher.onPlayerJoined(serverplayerentity, translatabletext.formatted(Formatting.YELLOW)).thenAccept(event -> {
+            if (!event.getPlayer().isConnected()) {
+                return;
+            }
+
+            event.getMessage().ifPresent(message -> this.broadcastChatMessage(org.loomdev.loom.util.transformer.TextTransformer.toMinecraft(message), MessageType.SYSTEM, Util.NIL_UUID));
         serverplaynetworkhandler.requestTeleport(serverplayerentity.getX(), serverplayerentity.getY(), serverplayerentity.getZ(), serverplayerentity.yaw, serverplayerentity.pitch);
         this.players.add(serverplayerentity);
         this.playerMap.put(serverplayerentity.getUuid(), serverplayerentity);
@@ -266,6 +272,8 @@
         }
 
         serverplayerentity.onSpawn();
+        });
+        // Loom end
     }
 
     protected void sendScoreboard(ServerScoreboard serverscoreboard, ServerPlayerEntity serverplayerentity) {
@@ -403,6 +411,10 @@
     public Text checkCanJoin(SocketAddress socketaddress, GameProfile gameprofile) {
         TranslatableText translatabletext;
 
+        // Loom start :: fire PlayerLoggedInEvent
+        // TODO
+        // Loom end
+
         if (this.bannedProfiles.contains(gameprofile)) {
             BannedPlayerEntry bannedplayerentry = (BannedPlayerEntry) this.bannedProfiles.get(gameprofile);
 
@@ -651,7 +663,7 @@
     }
 
     public void removeFromOperators(GameProfile gameprofile) {
-        this.ops.remove((Object) gameprofile);
+        this.ops.remove(gameprofile); // Loom :: decompile error
         ServerPlayerEntity serverplayerentity = this.getPlayer(gameprofile.getId());
 
         if (serverplayerentity != null) {
