--- a/net/minecraft/server/ServerResources.java
+++ b/net/minecraft/server/ServerResources.java
@@ -22,7 +22,7 @@
 
     private static final CompletableFuture<Unit> DATA_RELOAD_INITIAL_TASK = CompletableFuture.completedFuture(Unit.INSTANCE);
     private final ReloadableResourceManager resources;
-    private final Commands commands;
+    public final Commands commands; // Loom :: private -> public
     private final RecipeManager recipes;
     private final TagManager tagManager;
     private final PredicateManager predicateManager;
@@ -48,6 +48,7 @@
         this.resources.registerReloadListener(this.itemModifierManager);
         this.resources.registerReloadListener(this.functionLibrary);
         this.resources.registerReloadListener(this.advancements);
+        if (org.loomdev.api.Loom.getServer() != null) ((org.loomdev.loom.command.CommandManagerImpl) org.loomdev.api.Loom.getCommandManager()).internalReload(); // Loom
     }
 
     public ServerFunctionLibrary getFunctionLibrary() {
